// @ant-design/react-native/lib/image-picker/ImageItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var ImageItem = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ImageItem, _Component);

  var _super = (0, _createSuper2["default"])(ImageItem);

  function ImageItem(props) {
    (0, _classCallCheck2["default"])(this, ImageItem);
    return _super.call(this, props);
  }

  (0, _createClass2["default"])(ImageItem, [{
    key: "UNSAFE_componentWillMount",
    value: function UNSAFE_componentWillMount() {
      var _Dimensions$get = _reactNative.Dimensions.get('window'),
          width = _Dimensions$get.width;

      var _this$props = this.props,
          imageMargin = _this$props.imageMargin,
          imagesPerRow = _this$props.imagesPerRow,
          containerWidth = _this$props.containerWidth;

      if (typeof containerWidth !== 'undefined') {
        width = containerWidth;
      }

      this._imageSize = (width - (imagesPerRow + 1) * imageMargin) / imagesPerRow;
    }
  }, {
    key: "render",
    value: function render() {
      var _this = this;

      var _this$props2 = this.props,
          item = _this$props2.item,
          selected = _this$props2.selected,
          selectedMarker = _this$props2.selectedMarker,
          imageMargin = _this$props2.imageMargin;

      if (!item) {
        return null;
      }

      var marker = selectedMarker ? selectedMarker : /*#__PURE__*/_react["default"].createElement(_icon["default"], {
        name: "check-circle",
        style: [styles.marker]
      });
      var image = item.node.image;
      return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableOpacity, {
        style: {
          marginBottom: imageMargin,
          marginRight: imageMargin
        },
        onPress: function onPress() {
          return _this._handleClick(image);
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Image, {
        source: {
          uri: image.uri
        },
        style: {
          height: this._imageSize,
          width: this._imageSize
        }
      }), selected ? marker : null);
    }
  }, {
    key: "_handleClick",
    value: function _handleClick(item) {
      if (this.props.onPress) {
        this.props.onPress(item);
      }
    }
  }]);
  return ImageItem;
}(_react.Component);

ImageItem.defaultProps = {
  item: {},
  selected: false
};

var styles = _reactNative.StyleSheet.create({
  marker: {
    position: 'absolute',
    top: 5,
    right: 5,
    backgroundColor: 'transparent'
  }
});

var _default = ImageItem;
exports["default"] = _default;
