// @ant-design/react-native/es/result/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Image, ScrollView, Text } from 'react-native';
import { Result } from '../../';

var ResultExample = /*#__PURE__*/function (_React$Component) {
  _inherits(ResultExample, _React$Component);

  var _super = _createSuper(ResultExample);

  function ResultExample() {
    _classCallCheck(this, ResultExample);

    return _super.apply(this, arguments);
  }

  _createClass(ResultExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(ScrollView, {
        style: {
          backgroundColor: '#F5F5F9',
          flex: 1
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "URI \u56FE\u7247"), /*#__PURE__*/React.createElement(Result, {
        imgUrl: {
          uri: 'https://zos.alipayobjects.com/rmsportal/GcBguhrOdlYvGfnsXgrE.png'
        },
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1"
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "Image source"), /*#__PURE__*/React.createElement(Result, {
        imgUrl: require('./alipay.png'),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1"
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "Image element"), /*#__PURE__*/React.createElement(Result, {
        img: /*#__PURE__*/React.createElement(Image, {
          source: require('./alipay.png'),
          style: {
            width: 60,
            height: 60
          }
        }),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: /*#__PURE__*/React.createElement(Text, null, "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1")
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "\u542B button \u64CD\u4F5C"), /*#__PURE__*/React.createElement(Result, {
        img: /*#__PURE__*/React.createElement(Image, {
          source: require('./alipay.png'),
          style: {
            width: 60,
            height: 60
          }
        }),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1",
        buttonText: "\u5B8C\u6210",
        buttonType: "primary",
        onButtonClick: function onButtonClick(e) {
          return alert(e.toString());
        }
      }));
    }
  }]);

  return ResultExample;
}(React.Component);

export { ResultExample as default };
