// @ant-design/react-native/lib/picker/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _antdMobileDemoData = require("antd-mobile-demo-data");

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var data = require('./data.json');

var CustomChildren = function CustomChildren(props) {
  return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableOpacity, {
    onPress: props.onPress
  }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
    style: {
      height: 36,
      paddingLeft: 15,
      flexDirection: 'row',
      alignItems: 'center'
    }
  }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
    style: {
      flex: 1
    }
  }, props.children), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
    style: {
      textAlign: 'right',
      color: '#888',
      marginRight: 15
    }
  }, props.extra)));
};

var PopupExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(PopupExample, _React$Component);

  var _super = (0, _createSuper2["default"])(PopupExample);

  function PopupExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, PopupExample);
    _this = _super.call(this, props);

    _this.onPress = function () {
      setTimeout(function () {
        _this.setState({
          data: _antdMobileDemoData.district
        });
      }, 500);
    };

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.state = {
      data: [],
      value: [],
      pickerValue: []
    };
    return _this;
  }

  (0, _createClass2["default"])(PopupExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_.List, null, /*#__PURE__*/_react["default"].createElement(_.Picker, {
        data: data,
        cols: 3,
        value: this.state.value,
        onChange: this.onChange
      }, /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        arrow: "horizontal"
      }, "\u7701\u5E02\u9009\u62E9")), /*#__PURE__*/_react["default"].createElement(_.Picker, {
        data: this.state.data,
        cols: 2,
        value: this.state.value,
        onChange: this.onChange
      }, /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        arrow: "horizontal",
        onPress: this.onPress
      }, "\u7701\u5E02\u9009\u62E9(\u5F02\u6B65\u52A0\u8F7D)")), /*#__PURE__*/_react["default"].createElement(_.Picker, {
        title: "\u9009\u62E9\u5730\u533A",
        data: _antdMobileDemoData.district,
        cols: 2,
        value: this.state.pickerValue,
        onChange: function onChange(v) {
          return _this2.setState({
            pickerValue: v
          });
        },
        onOk: function onOk(v) {
          return _this2.setState({
            pickerValue: v
          });
        }
      }, /*#__PURE__*/_react["default"].createElement(CustomChildren, null, "Customized children"))));
    }
  }]);
  return PopupExample;
}(_react["default"].Component);

exports["default"] = PopupExample;
