// @ant-design/react-native/es/picker/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var data = require('./data.json');

import { district } from 'antd-mobile-demo-data';
import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { List, Picker } from '../../';

var CustomChildren = function CustomChildren(props) {
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: props.onPress
  }, /*#__PURE__*/React.createElement(View, {
    style: {
      height: 36,
      paddingLeft: 15,
      flexDirection: 'row',
      alignItems: 'center'
    }
  }, /*#__PURE__*/React.createElement(Text, {
    style: {
      flex: 1
    }
  }, props.children), /*#__PURE__*/React.createElement(Text, {
    style: {
      textAlign: 'right',
      color: '#888',
      marginRight: 15
    }
  }, props.extra)));
};

var PopupExample = /*#__PURE__*/function (_React$Component) {
  _inherits(PopupExample, _React$Component);

  var _super = _createSuper(PopupExample);

  function PopupExample(props) {
    var _this;

    _classCallCheck(this, PopupExample);

    _this = _super.call(this, props);

    _this.onPress = function () {
      setTimeout(function () {
        _this.setState({
          data: district
        });
      }, 500);
    };

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.state = {
      data: [],
      value: [],
      pickerValue: []
    };
    return _this;
  }

  _createClass(PopupExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/React.createElement(List, null, /*#__PURE__*/React.createElement(Picker, {
        data: data,
        cols: 3,
        value: this.state.value,
        onChange: this.onChange
      }, /*#__PURE__*/React.createElement(List.Item, {
        arrow: "horizontal"
      }, "\u7701\u5E02\u9009\u62E9")), /*#__PURE__*/React.createElement(Picker, {
        data: this.state.data,
        cols: 2,
        value: this.state.value,
        onChange: this.onChange
      }, /*#__PURE__*/React.createElement(List.Item, {
        arrow: "horizontal",
        onPress: this.onPress
      }, "\u7701\u5E02\u9009\u62E9(\u5F02\u6B65\u52A0\u8F7D)")), /*#__PURE__*/React.createElement(Picker, {
        title: "\u9009\u62E9\u5730\u533A",
        data: district,
        cols: 2,
        value: this.state.pickerValue,
        onChange: function onChange(v) {
          return _this2.setState({
            pickerValue: v
          });
        },
        onOk: function onOk(v) {
          return _this2.setState({
            pickerValue: v
          });
        }
      }, /*#__PURE__*/React.createElement(CustomChildren, null, "Customized children"))));
    }
  }]);

  return PopupExample;
}(React.Component);

export { PopupExample as default };
