// @ant-design/react-native/lib/stepper/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
function onChange(value) {
  console.log('changed', value);
}

var StepperExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(StepperExample, _React$Component);

  var _super = (0, _createSuper2["default"])(StepperExample);

  function StepperExample() {
    (0, _classCallCheck2["default"])(this, StepperExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(StepperExample, [{
    key: "render",
    value: function render() {
      var readOnly = /*#__PURE__*/_react["default"].createElement(_.Stepper, {
        key: "1",
        max: 10,
        min: 1,
        readOnly: false,
        defaultValue: 1,
        onChange: onChange
      });

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_.List, null, /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        extra: /*#__PURE__*/_react["default"].createElement(_.Stepper, {
          key: "0",
          max: 10,
          min: 1,
          defaultValue: 3,
          onChange: onChange
        })
      }, "readOnly: true"), /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        extra: readOnly
      }, "readOnly: false"), /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        extra: /*#__PURE__*/_react["default"].createElement(_.Stepper, {
          key: "2",
          disabled: true,
          max: 10,
          min: 1,
          defaultValue: 3,
          onChange: onChange
        })
      }, "Disabled")));
    }
  }]);
  return StepperExample;
}(_react["default"].Component);

exports["default"] = StepperExample;
