// @ant-design/react-native/es/tag/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { View } from 'react-native';
import { Tag, WhiteSpace } from '../../';

function onChange(selected) {
  console.log("tag selected: ".concat(selected));
}

var BasicTagExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicTagExample, _React$Component);

  var _super = _createSuper(BasicTagExample);

  function BasicTagExample() {
    _classCallCheck(this, BasicTagExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicTagExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/React.createElement(Tag, null, "Basic"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        selected: true
      }, "Selected"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        disabled: true
      }, "Disabled"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        onChange: onChange
      }, "Callback"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        closable: true,
        onClose: function onClose() {
          console.log('onClose');
        },
        afterClose: function afterClose() {
          console.log('afterClose');
        }
      }, "Closable"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        small: true
      }, "Small and Readonly"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Tag, {
        onLongPress: function onLongPress() {
          console.log('onLongPress');
        }
      }, "LongPress"));
    }
  }]);

  return BasicTagExample;
}(React.Component);

export { BasicTagExample as default };
