// @ant-design/react-native/lib/radio/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var RadioItem = _.Radio.RadioItem;

var BasicRadioExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicRadioExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicRadioExample);

  function BasicRadioExample() {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicRadioExample);
    _this = _super.apply(this, arguments);
    _this.state = {
      part1Value: 1,
      part2Value: 1
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicRadioExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Radio, {
        checked: this.state.part1Value === 1,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part1Value: 1
            });
          }
        },
        style: {
          borderWidth: 1,
          borderColor: '#999',
          margin: 10
        }
      }, "Support"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Radio, {
        checked: this.state.part1Value === 2,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part1Value: 2
            });
          }
        },
        style: {
          borderWidth: 1,
          borderColor: '#999',
          margin: 10
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null)), /*#__PURE__*/_react["default"].createElement(_.List, {
        style: {
          marginTop: 12
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          marginTop: 12
        }
      }, "Form radio, radio in general list."), /*#__PURE__*/_react["default"].createElement(RadioItem, {
        checked: this.state.part2Value === 1,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part2Value: 1
            });
          }
        }
      }, "Use Ant Design Component"), /*#__PURE__*/_react["default"].createElement(RadioItem, {
        checked: this.state.part2Value === 2,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part2Value: 2
            });
          }
        }
      }, "Use Ant Design Component"), /*#__PURE__*/_react["default"].createElement(RadioItem, {
        disabled: true
      }, "Set disabled"), /*#__PURE__*/_react["default"].createElement(RadioItem, {
        disabled: true,
        checked: true
      }, "Set disabled")));
    }
  }]);
  return BasicRadioExample;
}(_react["default"].Component);

exports["default"] = BasicRadioExample;
