// @ant-design/react-native/lib/steps/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var Step = _.Steps.Step;

var BasicTimelineExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicTimelineExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicTimelineExample);

  function BasicTimelineExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicTimelineExample);
    _this = _super.call(this, props);
    _this.state = {
      steps1: [{
        title: 'Finished',
        description: 'This is description'
      }, {
        title: 'In Progress',
        description: 'This is description'
      }, {
        title: 'Waiting',
        description: 'This is description'
      }],
      steps2: [{
        title: 'Finished',
        description: 'This is description',
        status: 'finish'
      }, {
        title: 'In Progress',
        description: 'This is description',
        status: 'process'
      }, {
        title: 'Waiting',
        description: 'This is description',
        status: 'error'
      }, {
        title: 'Waiting',
        description: 'This is description',
        status: 'wait'
      }]
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicTimelineExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
        style: {
          flex: 1
        },
        automaticallyAdjustContentInsets: false,
        showsHorizontalScrollIndicator: false,
        showsVerticalScrollIndicator: false
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 60
        }
      }, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        size: "small",
        current: 1,
        direction: "horizontal"
      }, this.state.steps1.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(Step, {
          key: index,
          title: /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "title:", item.title)),
          status: item.status
        });
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 60
        }
      }, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        size: "small",
        current: 1
      }, this.state.steps1.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(Step, {
          key: index,
          title: /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "title:", item.title)),
          description: /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "desc:", item.description)),
          status: item.status
        });
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        size: "small"
      }, this.state.steps2.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(Step, {
          key: index,
          title: item.title,
          description: item.description,
          status: item.status
        });
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        current: 1
      }, this.state.steps1.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(Step, {
          key: index,
          title: item.title,
          description: item.description,
          status: item.status
        });
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, null, this.state.steps2.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(Step, {
          key: index,
          title: item.title,
          description: item.description,
          status: item.status
        });
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        current: 1
      }, /*#__PURE__*/_react["default"].createElement(Step, {
        key: 0,
        title: "Finished",
        description: "This is description",
        status: "finish"
      }), /*#__PURE__*/_react["default"].createElement(Step, {
        key: 1,
        title: "Progress",
        description: "This is description",
        status: "progress"
      }), /*#__PURE__*/_react["default"].createElement(Step, {
        key: 2,
        title: "Wait",
        description: "This is description",
        status: "wait",
        icon: /*#__PURE__*/_react["default"].createElement(_.Icon, {
          name: "down",
          size: 20,
          color: "white"
        })
      })))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Steps, {
        current: 1
      }, /*#__PURE__*/_react["default"].createElement(Step, {
        key: 0,
        title: "Finished",
        description: "This is description",
        status: "finish",
        renderIcon: function renderIcon(_ref) {
          var starting = _ref.starting,
              waiting = _ref.waiting,
              error = _ref.error;
          var icon;

          if (starting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "home"
            });
          } else if (waiting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "user"
            });
          } else if (error) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "star"
            });
          }

          return icon;
        }
      }), /*#__PURE__*/_react["default"].createElement(Step, {
        key: 1,
        title: "Progress",
        description: "This is description",
        status: "progress",
        renderIcon: function renderIcon(_ref2) {
          var starting = _ref2.starting,
              waiting = _ref2.waiting,
              error = _ref2.error;
          var icon;

          if (starting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "home"
            });
          } else if (waiting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "user"
            });
          } else if (error) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "star"
            });
          }

          return icon;
        }
      }), /*#__PURE__*/_react["default"].createElement(Step, {
        key: 2,
        title: "Wait",
        description: "This is description",
        status: "wait",
        renderIcon: function renderIcon(_ref3) {
          var starting = _ref3.starting,
              waiting = _ref3.waiting,
              error = _ref3.error;
          var icon;

          if (starting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "home"
            });
          } else if (waiting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "user"
            });
          } else if (error) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "star"
            });
          }

          return icon;
        }
      }), /*#__PURE__*/_react["default"].createElement(Step, {
        key: 3,
        title: "Wait",
        description: "This is description",
        status: "error",
        renderIcon: function renderIcon(_ref4) {
          var starting = _ref4.starting,
              waiting = _ref4.waiting,
              error = _ref4.error;
          var icon;

          if (starting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "home"
            });
          } else if (waiting) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "user"
            });
          } else if (error) {
            icon = /*#__PURE__*/_react["default"].createElement(_.Icon, {
              name: "star"
            });
          }

          return icon;
        }
      })))));
    }
  }]);
  return BasicTimelineExample;
}(_react["default"].Component);

exports["default"] = BasicTimelineExample;
