// @ant-design/react-native/lib/wing-blank/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var WingBlank = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(WingBlank, _React$Component);

  var _super = (0, _createSuper2["default"])(WingBlank);

  function WingBlank() {
    (0, _classCallCheck2["default"])(this, WingBlank);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(WingBlank, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          style = _this$props.style,
          children = _this$props.children;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, null, function (_, theme) {
        var margin = theme["h_spacing_".concat(size)];
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [{
            marginLeft: margin,
            marginRight: margin
          }, style]
        }, children);
      });
    }
  }]);
  return WingBlank;
}(_react["default"].Component);

WingBlank.defaultProps = {
  size: 'lg'
};
var _default = WingBlank;
exports["default"] = _default;
