// @ant-design/react-native/lib/locale-provider/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = _interopRequireDefault(require(".."));

var _2 = require("../../");

var _en_US = _interopRequireDefault(require("../en_US"));

var _es_ES = _interopRequireDefault(require("../es_ES"));

var _pt_BR = _interopRequireDefault(require("../pt_BR"));

var _ru_RU = _interopRequireDefault(require("../ru_RU"));

var _zh_CN = _interopRequireDefault(require("../zh_CN"));

var maxDate = new Date(2018, 11, 3, 22, 0);
var minDate = new Date(2015, 7, 6, 8, 30);
var seasons = [[{
  label: '2013',
  value: '2013'
}, {
  label: '2014',
  value: '2014'
}], [{
  label: '春',
  value: '春'
}, {
  label: '夏',
  value: '夏'
}]];

var Page = /*#__PURE__*/_react["default"].memo(function () {
  return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_2.Pagination, {
    total: 5,
    current: 1
  }), /*#__PURE__*/_react["default"].createElement(_2.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_2.List, {
    style: {
      backgroundColor: 'white'
    }
  }, /*#__PURE__*/_react["default"].createElement(_2.DatePicker, {
    mode: "date",
    title: "Select date",
    minDate: minDate,
    maxDate: maxDate
  }, /*#__PURE__*/_react["default"].createElement(_2.List.Item, {
    arrow: "horizontal"
  }, "DatePicker")), /*#__PURE__*/_react["default"].createElement(_2.Picker, {
    data: seasons,
    cascade: false
  }, /*#__PURE__*/_react["default"].createElement(_2.List.Item, {
    arrow: "horizontal"
  }, "Picker")), /*#__PURE__*/_react["default"].createElement(_2.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_2.SearchBar, {
    placeholder: "Search",
    showCancelButton: true
  })));
});

var LocaleProviderExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(LocaleProviderExample, _React$Component);

  var _super = (0, _createSuper2["default"])(LocaleProviderExample);

  function LocaleProviderExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, LocaleProviderExample);
    _this = _super.call(this, props);

    _this.onChange = function (value) {
      _this.setState({
        locale: value[0]
      });
    };

    _this.state = {
      locale: 'English'
    };
    return _this;
  }

  (0, _createClass2["default"])(LocaleProviderExample, [{
    key: "render",
    value: function render() {
      var locale = this.state.locale;
      var languages = [{
        value: '中国',
        label: '中国',
        language: _zh_CN["default"]
      }, {
        value: 'English',
        label: 'English',
        language: _en_US["default"]
      }, {
        value: 'Русский',
        label: 'Русский',
        language: _ru_RU["default"]
      }, {
        value: 'Español',
        label: 'Español',
        language: _es_ES["default"]
      }, {
        value: 'Português - BR',
        label: 'Português - BR',
        language: _pt_BR["default"]
      }];
      var currentLocale = languages.find(function (item) {
        return item.value === locale;
      }).language;
      return /*#__PURE__*/_react["default"].createElement(_2.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_2.Picker, {
        data: languages,
        onChange: this.onChange,
        cols: 1,
        value: [locale]
      }, /*#__PURE__*/_react["default"].createElement(_2.List.Item, {
        arrow: "horizontal"
      }, "Choose language")), /*#__PURE__*/_react["default"].createElement(_2.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_["default"], {
        locale: currentLocale
      }, /*#__PURE__*/_react["default"].createElement(Page, null)));
    }
  }]);
  return LocaleProviderExample;
}(_react["default"].Component);

exports["default"] = LocaleProviderExample;
