// @ant-design/react-native/lib/swipe-action/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var BasicSwipeActionExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicSwipeActionExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicSwipeActionExample);

  function BasicSwipeActionExample() {
    (0, _classCallCheck2["default"])(this, BasicSwipeActionExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicSwipeActionExample, [{
    key: "render",
    value: function render() {
      var right = [{
        text: 'More',
        onPress: function onPress() {
          return console.log('more');
        },
        backgroundColor: 'orange',
        color: 'white'
      }, {
        text: 'Delete',
        onPress: function onPress() {
          return console.log('delete');
        },
        backgroundColor: 'red',
        color: 'white'
      }];
      var left = [{
        text: 'Read',
        onPress: function onPress() {
          return console.log('read');
        },
        backgroundColor: 'blue',
        color: 'white'
      }, {
        text: 'Reply',
        onPress: function onPress() {
          return console.log('reply');
        },
        backgroundColor: 'green',
        color: 'white'
      }];
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_.List, null, /*#__PURE__*/_react["default"].createElement(_.SwipeAction, {
        right: right,
        left: left,
        onSwipeableOpen: function onSwipeableOpen() {
          return console.log('open');
        },
        onSwipeableClose: function onSwipeableClose() {
          return console.log('close');
        }
      }, /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        extra: "extra content"
      }, "Simple example: left and right buttons"))));
    }
  }]);
  return BasicSwipeActionExample;
}(_react["default"].Component);

exports["default"] = BasicSwipeActionExample;
