// @ant-design/react-native/lib/action-sheet/AndroidContainer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ModalView = _interopRequireDefault(require("../modal/ModalView"));

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var ActionSheetAndroid = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2["default"])(ActionSheetAndroid, _React$PureComponent);

  var _super = (0, _createSuper2["default"])(ActionSheetAndroid);

  function ActionSheetAndroid(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ActionSheetAndroid);
    _this = _super.call(this, props);

    _this.close = function () {
      _this.setState({
        visible: false
      });
    };

    _this.state = {
      visible: _this.props.visible || false
    };
    return _this;
  }

  (0, _createClass2["default"])(ActionSheetAndroid, [{
    key: "confirm",
    value: function confirm(index) {
      var callback = this.props.callback;

      if (callback) {
        callback(index);
      }

      this.setState({
        visible: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          config = _this$props.config,
          onAnimationEnd = _this$props.onAnimationEnd;
      var title = config.title,
          message = config.message,
          options = config.options,
          destructiveButtonIndex = config.destructiveButtonIndex,
          cancelButtonIndex = config.cancelButtonIndex;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        themeStyles: _index["default"],
        styles: this.props.styles
      }, function (styles, theme) {
        var titleMsg = !!title && /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.title,
          key: "0"
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: styles.titleText
        }, title));

        var content = options.map(function (item, index) {
          return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            key: index,
            style: [cancelButtonIndex === index ? styles.cancelBtn : undefined]
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableHighlight, {
            style: [styles.btn],
            underlayColor: theme.fill_tap,
            onPress: function onPress() {
              return _this2.confirm(index);
            }
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.btnText, destructiveButtonIndex === index ? styles.destructiveBtn : undefined]
          }, item)), cancelButtonIndex === index ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.cancelBtnMask
          }) : null);
        });
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.container
        }, /*#__PURE__*/_react["default"].createElement(_ModalView["default"], {
          animationDuration: 200,
          animateAppear: true,
          visible: _this2.state.visible,
          onAnimationEnd: onAnimationEnd,
          style: styles.content,
          animationType: "slide-up",
          maskClosable: true,
          onClose: function onClose() {
            return _this2.confirm(cancelButtonIndex || -1);
          }
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, titleMsg, !!message && /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.message,
          key: "1"
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, message)), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, content))));
      });
    }
  }]);
  return ActionSheetAndroid;
}(_react["default"].PureComponent);

var _default = ActionSheetAndroid;
exports["default"] = _default;
