// @ant-design/react-native/es/date-picker-view/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { Text, View } from 'react-native';
import { DatePickerView } from '../../';

var DatePickerViewExample = /*#__PURE__*/function (_React$Component) {
  _inherits(DatePickerViewExample, _React$Component);

  var _super = _createSuper(DatePickerViewExample);

  function DatePickerViewExample() {
    var _this;

    _classCallCheck(this, DatePickerViewExample);

    _this = _super.apply(this, arguments);
    _this.state = {
      value: undefined,
      value12hours: undefined
    };

    _this.onChange = function (value) {
      console.log(value);

      _this.setState({
        value: value
      });
    };

    _this.onValueChange = function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      console.log(args);
    };

    return _this;
  }

  _createClass(DatePickerViewExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 16
        }
      }, "use12Hours"), /*#__PURE__*/React.createElement(DatePickerView, {
        value: this.state.value12hours,
        onChange: function onChange(v) {
          return _this2.setState({
            value12hours: v
          });
        },
        use12Hours: true
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 16
        }
      }, "Start DateTime"), /*#__PURE__*/React.createElement(DatePickerView, {
        value: this.state.value,
        onChange: this.onChange,
        onValueChange: this.onValueChange
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 16
        }
      }, "End DateTime"), /*#__PURE__*/React.createElement(DatePickerView, {
        value: this.state.value,
        onChange: this.onChange,
        onValueChange: this.onValueChange
      }));
    }
  }]);

  return DatePickerViewExample;
}(React.Component);

export { DatePickerViewExample as default };
