// @ant-design/react-native/es/image-picker/index.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Image, Text, TouchableOpacity, TouchableWithoutFeedback, View } from 'react-native';
import { WithTheme } from '../style';
import ImageRoll from './ImageRoll';
import ImagePickerStyles from './style/index';

var ImagePicker = /*#__PURE__*/function (_React$Component) {
  _inherits(ImagePicker, _React$Component);

  var _super = _createSuper(ImagePicker);

  function ImagePicker(props) {
    var _this;

    _classCallCheck(this, ImagePicker);

    _this = _super.call(this, props);

    _this.onPressIn = function (styles) {
      return function () {
        _this.plusWrap.setNativeProps({
          style: [styles.item, styles.size, styles.plusWrapHighlight]
        });
      };
    };

    _this.onPressOut = function (styles) {
      return function () {
        _this.plusWrap.setNativeProps({
          style: [styles.item, styles.size, styles.plusWrapNormal]
        });
      };
    };

    _this.showPicker = function () {
      if (_this.props.onAddImageClick) {
        _this.props.onAddImageClick();

        return;
      }

      _this.setState({
        visible: true
      });
    };

    _this.hideImageRoll = function () {
      _this.setState({
        visible: false
      });

      if (_this.props.onFail) {
        _this.props.onFail('cancel image selection');
      }
    };

    _this.state = {
      visible: false
    };
    return _this;
  }

  _createClass(ImagePicker, [{
    key: "addImage",
    value: function addImage(imageObj) {
      if (!imageObj.url) {
        imageObj.url = imageObj.uri;
        delete imageObj.uri;
      }

      var _this$props$files = this.props.files,
          files = _this$props$files === void 0 ? [] : _this$props$files;
      var newImages = files.concat(imageObj);

      if (this.props.onChange) {
        this.props.onChange(newImages, 'add');
      }
    }
  }, {
    key: "removeImage",
    value: function removeImage(idx) {
      var newImages = [];
      var _this$props$files2 = this.props.files,
          files = _this$props$files2 === void 0 ? [] : _this$props$files2;
      files.forEach(function (image, index) {
        if (index !== idx) {
          newImages.push(image);
        }
      });

      if (this.props.onChange) {
        this.props.onChange(newImages, 'remove', idx);
      }
    }
  }, {
    key: "onImageClick",
    value: function onImageClick(index) {
      if (this.props.onImageClick) {
        this.props.onImageClick(index, this.props.files);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          _this$props$files3 = _this$props.files,
          files = _this$props$files3 === void 0 ? [] : _this$props$files3,
          selectable = _this$props.selectable,
          cameraPickerProps = _this$props.cameraPickerProps;
      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: ImagePickerStyles
      }, function (styles) {
        var filesView = files.map(function (item, index) {
          return /*#__PURE__*/React.createElement(View, {
            key: index,
            style: [styles.item, styles.size]
          }, /*#__PURE__*/React.createElement(TouchableOpacity, {
            onPress: function onPress() {
              return _this2.onImageClick(index);
            },
            activeOpacity: 0.6
          }, /*#__PURE__*/React.createElement(Image, {
            source: {
              uri: item.url
            },
            style: [styles.size, styles.image]
          })), /*#__PURE__*/React.createElement(TouchableOpacity, {
            onPress: function onPress() {
              return _this2.removeImage(index);
            },
            style: styles.closeWrap,
            activeOpacity: 0.6
          }, /*#__PURE__*/React.createElement(Text, {
            style: styles.closeText
          }, "\xD7")));
        });
        var imageRollEl = /*#__PURE__*/React.createElement(ImageRoll, {
          onCancel: _this2.hideImageRoll,
          onSelected: function onSelected(imgObj) {
            return _this2.addImage(imgObj);
          },
          title: _this2.props.title,
          cancelText: _this2.props.cancelText,
          cameraPickerProps: cameraPickerProps
        });
        return /*#__PURE__*/React.createElement(View, {
          style: styles.container
        }, filesView, selectable && /*#__PURE__*/React.createElement(TouchableWithoutFeedback, {
          onPress: _this2.showPicker,
          onPressIn: _this2.onPressIn(styles),
          onPressOut: _this2.onPressOut(styles)
        }, /*#__PURE__*/React.createElement(View, {
          ref: function ref(conponent) {
            return _this2.plusWrap = conponent;
          },
          style: [styles.item, styles.size, styles.plusWrap, styles.plusWrapNormal]
        }, /*#__PURE__*/React.createElement(Text, {
          style: [styles.plusText]
        }, "+"))), _this2.state.visible ? imageRollEl : null);
      });
    }
  }]);

  return ImagePicker;
}(React.Component);

export { ImagePicker as default };
ImagePicker.defaultProps = {
  onChange: function onChange() {},
  onFail: function onFail() {},
  files: [],
  selectable: true
};
