// @ant-design/react-native/es/grid/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { ScrollView, Text, View } from 'react-native';
import { Grid } from '../../';
var data = Array.from(new Array(9)).map(function (_val, i) {
  return {
    icon: 'https://os.alipayobjects.com/rmsportal/IptWdCkrtkAUfjE.png',
    text: "Name".concat(i)
  };
});

var BasicGridExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicGridExample, _React$Component);

  var _super = _createSuper(BasicGridExample);

  function BasicGridExample() {
    _classCallCheck(this, BasicGridExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicGridExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(ScrollView, null, /*#__PURE__*/React.createElement(View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Simple")), /*#__PURE__*/React.createElement(View, {
        style: [{
          padding: 10
        }]
      }, /*#__PURE__*/React.createElement(Grid, {
        data: data,
        hasLine: false
      })), /*#__PURE__*/React.createElement(View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel")), /*#__PURE__*/React.createElement(Grid, {
        data: data,
        columnNum: 3,
        isCarousel: true,
        carouselProps: {
          style: {
            width: '100%',
            height: 320
          }
        },
        onPress: function onPress(_el, index) {
          return alert(index);
        }
      }), /*#__PURE__*/React.createElement(View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Custom GridCell Style")), /*#__PURE__*/React.createElement(Grid, {
        data: data,
        columnNum: 3,
        itemStyle: {
          height: 150,
          backgroundColor: '#ffff00'
        }
      }));
    }
  }]);

  return BasicGridExample;
}(React.Component);

export { BasicGridExample as default };
