// @ant-design/react-native/lib/slider/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var BasicSliderExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicSliderExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicSliderExample);

  function BasicSliderExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicSliderExample);
    _this = _super.call(this, props);

    _this.handleChange = function (value) {
      _this.setState({
        changingValue: value
      });
    };

    _this.onAfterChange = function (value) {
      _this.setState({
        changedValue: value
      });
    };

    _this.minMaxChange = function (value) {
      _this.setState({
        minMaxValue: value
      });
    };

    _this.state = {
      changingValue: 0.25,
      changedValue: 0.15,
      minMaxValue: 0,
      slideCompletionCount: 0
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicSliderExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Default settings"), /*#__PURE__*/_react["default"].createElement(_.Slider, null)), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Initial value: 0.5"), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        defaultValue: 0.5
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "min: 0, max: 1, current Value: ", this.state.minMaxValue), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        min: 0,
        max: 1,
        onAfterChange: function onAfterChange(value) {
          return _this2.minMaxChange(value);
        }
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "step: 0.25"), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        step: 0.25,
        value: 0.25
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "disabled"), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        disabled: true,
        defaultValue: 0.25
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "onChange value: ", this.state.changingValue), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        defaultValue: 0.25,
        onChange: function onChange(value) {
          return _this2.handleChange(value);
        }
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "onAfterChange value: ", this.state.changedValue), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        defaultValue: 0.15,
        onAfterChange: function onAfterChange(value) {
          return _this2.onAfterChange(value);
        }
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "custom color: "), /*#__PURE__*/_react["default"].createElement(_.Slider, {
        defaultValue: 0.15,
        minimumTrackTintColor: "red",
        maximumTrackTintColor: "blue"
      })));
    }
  }]);
  return BasicSliderExample;
}(_react["default"].Component);

exports["default"] = BasicSliderExample;
