// @ant-design/react-native/es/date-picker/utils.js
import _typeof from "@babel/runtime/helpers/esm/typeof";

function formatIt(date, form) {
  var pad = function pad(n) {
    return n < 10 ? "0".concat(n) : n;
  };

  var dateStr = "".concat(date.getFullYear(), "-").concat(pad(date.getMonth() + 1), "-").concat(pad(date.getDate()));
  var timeStr = "".concat(pad(date.getHours()), ":").concat(pad(date.getMinutes()));

  if (form === 'YYYY-MM-DD') {
    return dateStr;
  }

  if (form === 'HH:mm') {
    return timeStr;
  }

  return "".concat(dateStr, " ").concat(timeStr);
}

export function formatFn(instance, value) {
  var formatsEnum = {
    date: 'YYYY-MM-DD',
    time: 'HH:mm',
    datetime: 'YYYY-MM-DD HH:mm'
  };
  var format = instance.props.format;

  var type = _typeof(format);

  if (type === 'string') {
    return formatIt(value, format);
  }

  if (type === 'function') {
    return format(value);
  }

  return formatIt(value, formatsEnum[instance.props.mode]);
}
export function formatProps(props, value) {
  var formatsEnum = {
    date: 'YYYY-MM-DD',
    time: 'HH:mm',
    datetime: 'YYYY-MM-DD HH:mm'
  };
  var format = props.format;

  var type = _typeof(format);

  if (type === 'string') {
    return formatIt(value, format);
  }

  if (type === 'function') {
    return format(value);
  }

  return formatIt(value, formatsEnum[props.mode]);
}
