// @ant-design/react-native/lib/action-sheet/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.description = exports.title = exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var Test = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Test, _React$Component);

  var _super = (0, _createSuper2["default"])(Test);

  function Test(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Test);
    _this = _super.call(this, props);

    _this.showActionSheet = function () {
      var BUTTONS = ['Operation1', 'Operation2', 'Operation3', 'Delete', 'Cancel'];

      _.ActionSheet.showActionSheetWithOptions({
        title: 'Title',
        message: 'Description',
        options: BUTTONS,
        cancelButtonIndex: 4,
        destructiveButtonIndex: 3
      }, function (buttonIndex) {
        _this.setState({
          clicked: BUTTONS[buttonIndex]
        });
      });
    };

    _this.showShareActionSheet = function () {
      var opts = {
        message: 'Message to go with the shared url',
        title: 'Share Actionsheet'
      };

      if (_reactNative.Platform.OS === 'ios') {
        opts.url = 'https://www.alipay.com/';
        opts.tintColor = '#ff0000';
        opts.excludedActivityTypes = ['com.apple.UIKit.activity.PostToTwitter'];
      }

      _.ActionSheet.showShareActionSheetWithOptions(opts, function (error) {
        return alert(error);
      }, function (success, method) {
        var text;

        if (success) {
          text = "Shared with ".concat(method);
        } else {
          text = 'Did not share';
        }

        _this.setState({
          text: text
        });
      });
    };

    _this.state = {
      clicked: 'none',
      text: ''
    };
    return _this;
  }

  (0, _createClass2["default"])(Test, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          padding: 8
        }]
      }, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.showActionSheet
      }, "showActionSheet")), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: [{
          padding: 8
        }]
      }, "clicked button: ", this.state.clicked), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          padding: 8
        }]
      }, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.showShareActionSheet
      }, "showShareActionSheet")), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: [{
          padding: 8
        }]
      }, this.state.text));
    }
  }]);
  return Test;
}(_react["default"].Component);

exports["default"] = Test;
var title = 'ActionSheet';
exports.title = title;
var description = 'ActionSheet example';
exports.description = description;
