// @ant-design/react-native/es/progress/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Button, Progress, WhiteSpace } from '../../';

var BasicProgressExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicProgressExample, _React$Component);

  var _super = _createSuper(BasicProgressExample);

  function BasicProgressExample(props) {
    var _this;

    _classCallCheck(this, BasicProgressExample);

    _this = _super.call(this, props);

    _this.onAdd = function () {
      var p = _this.state.percent + 10;

      if (_this.state.percent >= 100) {
        p = 0;
      }

      _this.setState({
        percent: p
      });
    };

    _this.state = {
      percent: 40
    };
    return _this;
  }

  _createClass(BasicProgressExample, [{
    key: "render",
    value: function render() {
      var style = {
        marginTop: 80,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
      };
      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Progress, {
        percent: 90,
        position: "fixed"
      }), /*#__PURE__*/React.createElement(View, {
        style: [style]
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          marginRight: 10,
          height: 4,
          flex: 1
        }
      }, /*#__PURE__*/React.createElement(Progress, {
        percent: this.state.percent
      })), /*#__PURE__*/React.createElement(Text, null, this.state.percent, "%")), /*#__PURE__*/React.createElement(Button, {
        style: {
          width: 50,
          marginLeft: 10
        },
        type: "ghost",
        size: "small",
        onPress: this.onAdd
      }, "(+-)10"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Progress, {
        percent: 5
      }));
    }
  }]);

  return BasicProgressExample;
}(React.Component);

export { BasicProgressExample as default };
