// @ant-design/react-native/es/checkbox/AgreeItem.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import { WithTheme } from '../style';
import Checkbox from './Checkbox';
import AgreeItemstyles from './style/index';

var AgreeItem = /*#__PURE__*/function (_React$Component) {
  _inherits(AgreeItem, _React$Component);

  var _super = _createSuper(AgreeItem);

  function AgreeItem() {
    var _this;

    _classCallCheck(this, AgreeItem);

    _this = _super.apply(this, arguments);

    _this.handleClick = function () {
      if (_this.checkbox) {
        _this.checkbox.handleClick();
      }
    };

    return _this;
  }

  _createClass(AgreeItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          checkboxStyle = _this$props.checkboxStyle,
          children = _this$props.children,
          disabled = _this$props.disabled,
          checked = _this$props.checked,
          defaultChecked = _this$props.defaultChecked,
          onChange = _this$props.onChange;
      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: AgreeItemstyles
      }, function (styles) {
        var contentDom = !children ? null : /*#__PURE__*/React.isValidElement(children) ? children : /*#__PURE__*/React.createElement(Text, null, children);
        return /*#__PURE__*/React.createElement(TouchableWithoutFeedback, {
          onPress: _this2.handleClick
        }, /*#__PURE__*/React.createElement(View, {
          style: [styles.agreeItem, style]
        }, /*#__PURE__*/React.createElement(Checkbox, {
          ref: function ref(_ref) {
            return _this2.checkbox = _ref;
          },
          style: [styles.agreeItemCheckbox, checkboxStyle],
          disabled: disabled,
          checked: checked,
          defaultChecked: defaultChecked,
          onChange: onChange
        }), /*#__PURE__*/React.createElement(View, {
          style: {
            flex: 1
          }
        }, contentDom)));
      });
    }
  }]);

  return AgreeItem;
}(React.Component);

export { AgreeItem as default };
