// @ant-design/react-native/es/checkbox/CheckboxItem.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import List from '../list/index';
import { WithTheme } from '../style';
import Checkbox from './Checkbox';
import CheckboxItemStyles from './style/index';
var ListItem = List.Item;

var CheckboxItem = /*#__PURE__*/function (_React$Component) {
  _inherits(CheckboxItem, _React$Component);

  var _super = _createSuper(CheckboxItem);

  function CheckboxItem() {
    var _this;

    _classCallCheck(this, CheckboxItem);

    _this = _super.apply(this, arguments);

    _this.handleClick = function () {
      if (_this.checkbox) {
        _this.checkbox.handleClick();
      }

      if (_this.props.onPress) {
        _this.props.onPress();
      }
    };

    return _this;
  }

  _createClass(CheckboxItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          checkboxStyle = _this$props.checkboxStyle,
          defaultChecked = _this$props.defaultChecked,
          checked = _this$props.checked,
          disabled = _this$props.disabled,
          children = _this$props.children,
          extra = _this$props.extra,
          onChange = _this$props.onChange;
      var thumbNode = /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: CheckboxItemStyles
      }, function (styles) {
        return /*#__PURE__*/React.createElement(Checkbox, {
          ref: function ref(_ref) {
            return _this2.checkbox = _ref;
          },
          style: [styles.checkboxItemCheckbox, checkboxStyle],
          defaultChecked: defaultChecked,
          checked: checked,
          onChange: onChange,
          disabled: disabled
        });
      });
      return /*#__PURE__*/React.createElement(ListItem, {
        style: style,
        onPress: disabled ? undefined : this.handleClick,
        extra: extra,
        thumb: thumbNode
      }, children);
    }
  }]);

  return CheckboxItem;
}(React.Component);

export { CheckboxItem as default };
