// @ant-design/react-native/es/checkbox/Checkbox.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import Icon from '../icon';
import { WithTheme } from '../style';
import CheckboxStyles from './style/index';

var Checkbox = /*#__PURE__*/function (_React$Component) {
  _inherits(Checkbox, _React$Component);

  var _super = _createSuper(Checkbox);

  function Checkbox(props, context) {
    var _this;

    _classCallCheck(this, Checkbox);

    _this = _super.call(this, props, context);

    _this.handleClick = function () {
      if (_this.props.disabled) {
        return;
      }

      var checked = !_this.state.checked;

      if (!(typeof _this.props.checked === 'boolean')) {
        _this.setState({
          checked: checked
        });
      }

      if (_this.props.onChange) {
        _this.props.onChange({
          target: {
            checked: checked
          }
        });
      }
    };

    _this.state = {
      checked: props.checked || props.defaultChecked || false
    };
    return _this;
  }

  _createClass(Checkbox, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (typeof nextProps.checked === 'boolean') {
        this.setState({
          checked: !!nextProps.checked
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          disabled = _this$props.disabled,
          children = _this$props.children;
      var checked = this.state.checked;
      return /*#__PURE__*/React.createElement(WithTheme, {
        themeStyles: CheckboxStyles,
        styles: this.props.styles
      }, function (styles, theme) {
        var icon;

        if (checked) {
          icon = disabled ? /*#__PURE__*/React.createElement(Icon, {
            name: "check-square",
            style: [styles.icon, style]
          }) : /*#__PURE__*/React.createElement(Icon, {
            name: "check-square",
            color: theme.brand_primary,
            style: [styles.icon, style]
          });
        } else {
          icon = disabled ? /*#__PURE__*/React.createElement(Icon, {
            name: "border",
            style: [styles.icon, style]
          }) : /*#__PURE__*/React.createElement(Icon, {
            name: "border",
            color: theme.brand_primary,
            style: [styles.icon, style]
          });
        }

        return /*#__PURE__*/React.createElement(TouchableWithoutFeedback, {
          onPress: _this2.handleClick
        }, /*#__PURE__*/React.createElement(View, {
          style: [styles.wrapper]
        }, icon, typeof children === 'string' ?
        /*#__PURE__*/
        // tslint:disable-next-line:jsx-no-multiline-js
        React.createElement(Text, {
          style: styles.iconRight
        }, _this2.props.children) : children));
      });
    }
  }]);

  return Checkbox;
}(React.Component);

export { Checkbox as default };
