// @ant-design/react-native/lib/portal/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.description = exports.title = exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../..");

var PopoverExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(PopoverExample, _React$Component);

  var _super = (0, _createSuper2["default"])(PopoverExample);

  function PopoverExample() {
    (0, _classCallCheck2["default"])(this, PopoverExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(PopoverExample, [{
    key: "render",
    value: function render() {
      var key;

      var contents = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          backgroundColor: 'green',
          padding: 100
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          _.Portal.remove(key);
        },
        type: "primary",
        style: {
          position: 'absolute',
          top: 100,
          right: 0
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Icon, {
        name: "close-circle"
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "\u81EA\u5B9A\u4E49Portal")));

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          key = _.Portal.add(contents);
        }
      }, "Open Portal"));
    }
  }]);
  return PopoverExample;
}(_react["default"].Component);

exports["default"] = PopoverExample;
var title = 'Portal';
exports.title = title;
var description = 'Portal example';
exports.description = description;
