// @ant-design/react-native/lib/date-picker-view/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var DatePickerViewExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(DatePickerViewExample, _React$Component);

  var _super = (0, _createSuper2["default"])(DatePickerViewExample);

  function DatePickerViewExample() {
    var _this;

    (0, _classCallCheck2["default"])(this, DatePickerViewExample);
    _this = _super.apply(this, arguments);
    _this.state = {
      value: undefined,
      value12hours: undefined
    };

    _this.onChange = function (value) {
      console.log(value);

      _this.setState({
        value: value
      });
    };

    _this.onValueChange = function () {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      console.log(args);
    };

    return _this;
  }

  (0, _createClass2["default"])(DatePickerViewExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 16
        }
      }, "use12Hours"), /*#__PURE__*/_react["default"].createElement(_.DatePickerView, {
        value: this.state.value12hours,
        onChange: function onChange(v) {
          return _this2.setState({
            value12hours: v
          });
        },
        use12Hours: true
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 16
        }
      }, "Start DateTime"), /*#__PURE__*/_react["default"].createElement(_.DatePickerView, {
        value: this.state.value,
        onChange: this.onChange,
        onValueChange: this.onValueChange
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 16
        }
      }, "End DateTime"), /*#__PURE__*/_react["default"].createElement(_.DatePickerView, {
        value: this.state.value,
        onChange: this.onChange,
        onValueChange: this.onValueChange
      }));
    }
  }]);
  return DatePickerViewExample;
}(_react["default"].Component);

exports["default"] = DatePickerViewExample;
