// @ant-design/react-native/es/tab-bar/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Icon, SearchBar, TabBar } from '../../';

var BasicTabBarExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicTabBarExample, _React$Component);

  var _super = _createSuper(BasicTabBarExample);

  function BasicTabBarExample(props) {
    var _this;

    _classCallCheck(this, BasicTabBarExample);

    _this = _super.call(this, props);
    _this.state = {
      selectedTab: 'redTab'
    };
    return _this;
  }

  _createClass(BasicTabBarExample, [{
    key: "renderContent",
    value: function renderContent(pageText) {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          flex: 1,
          alignItems: 'center',
          backgroundColor: 'white'
        }
      }, /*#__PURE__*/React.createElement(SearchBar, {
        placeholder: "Search",
        showCancelButton: true
      }), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 50
        }
      }, pageText));
    }
  }, {
    key: "onChangeTab",
    value: function onChangeTab(tabName) {
      this.setState({
        selectedTab: tabName
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(TabBar, {
        unselectedTintColor: "#949494",
        tintColor: "#33A3F4",
        barTintColor: "#f5f5f5"
      }, /*#__PURE__*/React.createElement(TabBar.Item, {
        title: "Life",
        icon: /*#__PURE__*/React.createElement(Icon, {
          name: "home"
        }),
        selected: this.state.selectedTab === 'blueTab',
        onPress: function onPress() {
          return _this2.onChangeTab('blueTab');
        }
      }, this.renderContent('Life Tab')), /*#__PURE__*/React.createElement(TabBar.Item, {
        icon: /*#__PURE__*/React.createElement(Icon, {
          name: "ordered-list"
        }),
        title: "Koubei",
        badge: 2,
        selected: this.state.selectedTab === 'redTab',
        onPress: function onPress() {
          return _this2.onChangeTab('redTab');
        }
      }, this.renderContent('Koubei Tab')), /*#__PURE__*/React.createElement(TabBar.Item, {
        icon: /*#__PURE__*/React.createElement(Icon, {
          name: "like"
        }),
        title: "Friend",
        selected: this.state.selectedTab === 'greenTab',
        onPress: function onPress() {
          return _this2.onChangeTab('greenTab');
        }
      }, this.renderContent('Friend Tab')), /*#__PURE__*/React.createElement(TabBar.Item, {
        icon: /*#__PURE__*/React.createElement(Icon, {
          name: "user"
        }),
        title: "My",
        selected: this.state.selectedTab === 'yellowTab',
        onPress: function onPress() {
          return _this2.onChangeTab('yellowTab');
        }
      }, this.renderContent('My Tab')));
    }
  }]);

  return BasicTabBarExample;
}(React.Component);

export { BasicTabBarExample as default };
