// @ant-design/react-native/lib/date-picker-view/date-picker-view.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _datepicker = _interopRequireDefault(require("../date-picker/datepicker"));

var _getLocale = require("../_util/getLocale");

var _localeProvider = require("../locale-provider");

var DatePickerView = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(DatePickerView, _React$Component);

  var _super = (0, _createSuper2["default"])(DatePickerView);

  function DatePickerView() {
    (0, _classCallCheck2["default"])(this, DatePickerView);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(DatePickerView, [{
    key: "render",
    value: function render() {
      // tslint:disable-next-line:no-this-assignment
      var props = this.props,
          context = this.context;
      var locale = (0, _getLocale.getComponentLocale)(props, context, 'DatePickerView', function () {
        return require('./locale/zh_CN');
      }); // DatePicker use `defaultDate`, maybe because there are PopupDatePicker inside? @yiminghe
      // Here Use `date` instead of `defaultDate`, make it controlled fully.

      return /*#__PURE__*/_react["default"].createElement(_datepicker["default"], (0, _extends2["default"])({}, props, {
        locale: locale,
        date: props.value,
        onDateChange: props.onChange,
        onValueChange: props.onValueChange,
        onScrollChange: props.onScrollChange
      }));
    }
  }]);
  return DatePickerView;
}(_react["default"].Component);

exports["default"] = DatePickerView;
DatePickerView.defaultProps = {
  mode: 'datetime',
  // extra: '请选择',
  minuteStep: 1,
  use12Hours: false
};
DatePickerView.contextType = _localeProvider.LocaleContext;
