// @ant-design/react-native/lib/button/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("../../");

/* tslint:disable:no-unused-variable */

/* tslint:disable:no-console */
var _default = function _default() {
  return /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, null, "default"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    disabled: true
  }, "default disabled"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "primary"
  }, "primary"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "primary",
    disabled: true
  }, "primary disabled"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "warning"
  }, "warning"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "warning",
    disabled: true
  }, "warning disabled"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    loading: true
  }, "loading button"), /*#__PURE__*/_react["default"].createElement(_.Button, {
    activeStyle: false
  }, "\u65E0\u70B9\u51FB\u53CD\u9988"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
    activeStyle: {
      backgroundColor: 'red'
    }
  }, "custom feedback style"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
    style: {
      marginTop: 20,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center'
    }
  }, /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "ghost"
  }, "ghost"), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "ghost",
    disabled: true
  }, "ghost disabled"), /*#__PURE__*/_react["default"].createElement(_.Button, {
    type: "ghost",
    size: "small"
  }, "ghost")));
};

exports["default"] = _default;
