// @ant-design/react-native/es/action-sheet/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { View, Text, Platform } from 'react-native';
import { ActionSheet, Button } from '../../';

var Test = /*#__PURE__*/function (_React$Component) {
  _inherits(Test, _React$Component);

  var _super = _createSuper(Test);

  function Test(props) {
    var _this;

    _classCallCheck(this, Test);

    _this = _super.call(this, props);

    _this.showActionSheet = function () {
      var BUTTONS = ['Operation1', 'Operation2', 'Operation3', 'Delete', 'Cancel'];
      ActionSheet.showActionSheetWithOptions({
        title: 'Title',
        message: 'Description',
        options: BUTTONS,
        cancelButtonIndex: 4,
        destructiveButtonIndex: 3
      }, function (buttonIndex) {
        _this.setState({
          clicked: BUTTONS[buttonIndex]
        });
      });
    };

    _this.showShareActionSheet = function () {
      var opts = {
        message: 'Message to go with the shared url',
        title: 'Share Actionsheet'
      };

      if (Platform.OS === 'ios') {
        opts.url = 'https://www.alipay.com/';
        opts.tintColor = '#ff0000';
        opts.excludedActivityTypes = ['com.apple.UIKit.activity.PostToTwitter'];
      }

      ActionSheet.showShareActionSheetWithOptions(opts, function (error) {
        return alert(error);
      }, function (success, method) {
        var text;

        if (success) {
          text = "Shared with ".concat(method);
        } else {
          text = 'Did not share';
        }

        _this.setState({
          text: text
        });
      });
    };

    _this.state = {
      clicked: 'none',
      text: ''
    };
    return _this;
  }

  _createClass(Test, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/React.createElement(View, {
        style: [{
          padding: 8
        }]
      }, /*#__PURE__*/React.createElement(Button, {
        onPress: this.showActionSheet
      }, "showActionSheet")), /*#__PURE__*/React.createElement(Text, {
        style: [{
          padding: 8
        }]
      }, "clicked button: ", this.state.clicked), /*#__PURE__*/React.createElement(View, {
        style: [{
          padding: 8
        }]
      }, /*#__PURE__*/React.createElement(Button, {
        onPress: this.showShareActionSheet
      }, "showShareActionSheet")), /*#__PURE__*/React.createElement(Text, {
        style: [{
          padding: 8
        }]
      }, this.state.text));
    }
  }]);

  return Test;
}(React.Component);

export { Test as default };
export var title = 'ActionSheet';
export var description = 'ActionSheet example';
