// @ant-design/react-native/lib/activity-indicator/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var ActivityIndicatorExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(ActivityIndicatorExample, _React$Component);

  var _super = (0, _createSuper2["default"])(ActivityIndicatorExample);

  function ActivityIndicatorExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ActivityIndicatorExample);
    _this = _super.call(this, props);
    _this.state = {
      animating: false
    };
    _this.loadingToast = _this.loadingToast.bind((0, _assertThisInitialized2["default"])(_this));
    return _this;
  }

  (0, _createClass2["default"])(ActivityIndicatorExample, [{
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      clearTimeout(this.closeTimer);
    }
  }, {
    key: "loadingToast",
    value: function loadingToast() {
      var _this2 = this;

      this.setState({
        animating: !this.state.animating
      });
      this.closeTimer = setTimeout(function () {
        _this2.setState({
          animating: !_this2.state.animating
        });
      }, 2000);
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.demo]
      }, /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Flex, null, /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Icon without text")), /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_.ActivityIndicator, null)))), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xl",
        style: {
          backgroundColor: '#fff'
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Flex, null, /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Icon with text")), /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_.ActivityIndicator, {
        text: "Loading..."
      })))), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xl",
        style: {
          backgroundColor: '#fff'
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Flex, null, /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Dark Background")), /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.darkBg]
      }, /*#__PURE__*/_react["default"].createElement(_.ActivityIndicator, {
        color: "#fff"
      }))))), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xl",
        style: {
          backgroundColor: '#fff'
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Flex, null, /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Large Size")), /*#__PURE__*/_react["default"].createElement(_.Flex.Item, null, /*#__PURE__*/_react["default"].createElement(_.ActivityIndicator, {
        size: "large"
      })))), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xl",
        style: {
          backgroundColor: '#fff'
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.loadingToast
      }, "Click to show Toast")), /*#__PURE__*/_react["default"].createElement(_.ActivityIndicator, {
        animating: this.state.animating,
        toast: true,
        size: "large",
        text: "Loading..."
      }));
    }
  }]);
  return ActivityIndicatorExample;
}(_react["default"].Component);

exports["default"] = ActivityIndicatorExample;

var styles = _reactNative.StyleSheet.create({
  demo: {
    marginTop: 20
  },
  darkBg: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 89,
    height: 89,
    backgroundColor: '#2B2F42'
  },
  gray: {
    backgroundColor: '#CCC'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 8
  }
});
