// @ant-design/react-native/es/stepper/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { View } from 'react-native';
import { List, Stepper } from '../../';

function onChange(value) {
  console.log('changed', value);
}

var StepperExample = /*#__PURE__*/function (_React$Component) {
  _inherits(StepperExample, _React$Component);

  var _super = _createSuper(StepperExample);

  function StepperExample() {
    _classCallCheck(this, StepperExample);

    return _super.apply(this, arguments);
  }

  _createClass(StepperExample, [{
    key: "render",
    value: function render() {
      var readOnly = /*#__PURE__*/React.createElement(Stepper, {
        key: "1",
        max: 10,
        min: 1,
        readOnly: false,
        defaultValue: 1,
        onChange: onChange
      });
      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(List, null, /*#__PURE__*/React.createElement(List.Item, {
        extra: /*#__PURE__*/React.createElement(Stepper, {
          key: "0",
          max: 10,
          min: 1,
          defaultValue: 3,
          onChange: onChange
        })
      }, "readOnly: true"), /*#__PURE__*/React.createElement(List.Item, {
        extra: readOnly
      }, "readOnly: false"), /*#__PURE__*/React.createElement(List.Item, {
        extra: /*#__PURE__*/React.createElement(Stepper, {
          key: "2",
          disabled: true,
          max: 10,
          min: 1,
          defaultValue: 3,
          onChange: onChange
        })
      }, "Disabled")));
    }
  }]);

  return StepperExample;
}(React.Component);

export { StepperExample as default };
