// @ant-design/react-native/lib/checkbox/AgreeItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _Checkbox = _interopRequireDefault(require("./Checkbox"));

var _index = _interopRequireDefault(require("./style/index"));

var AgreeItem = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(AgreeItem, _React$Component);

  var _super = (0, _createSuper2["default"])(AgreeItem);

  function AgreeItem() {
    var _this;

    (0, _classCallCheck2["default"])(this, AgreeItem);
    _this = _super.apply(this, arguments);

    _this.handleClick = function () {
      if (_this.checkbox) {
        _this.checkbox.handleClick();
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(AgreeItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          checkboxStyle = _this$props.checkboxStyle,
          children = _this$props.children,
          disabled = _this$props.disabled,
          checked = _this$props.checked,
          defaultChecked = _this$props.defaultChecked,
          onChange = _this$props.onChange;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        var contentDom = !children ? null : /*#__PURE__*/_react["default"].isValidElement(children) ? children : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, children);
        return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
          onPress: _this2.handleClick
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.agreeItem, style]
        }, /*#__PURE__*/_react["default"].createElement(_Checkbox["default"], {
          ref: function ref(_ref) {
            return _this2.checkbox = _ref;
          },
          style: [styles.agreeItemCheckbox, checkboxStyle],
          disabled: disabled,
          checked: checked,
          defaultChecked: defaultChecked,
          onChange: onChange
        }), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: {
            flex: 1
          }
        }, contentDom)));
      });
    }
  }]);
  return AgreeItem;
}(_react["default"].Component);

exports["default"] = AgreeItem;
