// @ant-design/react-native/lib/checkbox/CheckboxItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _index = _interopRequireDefault(require("../list/index"));

var _style = require("../style");

var _Checkbox = _interopRequireDefault(require("./Checkbox"));

var _index2 = _interopRequireDefault(require("./style/index"));

var ListItem = _index["default"].Item;

var CheckboxItem = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(CheckboxItem, _React$Component);

  var _super = (0, _createSuper2["default"])(CheckboxItem);

  function CheckboxItem() {
    var _this;

    (0, _classCallCheck2["default"])(this, CheckboxItem);
    _this = _super.apply(this, arguments);

    _this.handleClick = function () {
      if (_this.checkbox) {
        _this.checkbox.handleClick();
      }

      if (_this.props.onPress) {
        _this.props.onPress();
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(CheckboxItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          checkboxStyle = _this$props.checkboxStyle,
          defaultChecked = _this$props.defaultChecked,
          checked = _this$props.checked,
          disabled = _this$props.disabled,
          children = _this$props.children,
          extra = _this$props.extra,
          onChange = _this$props.onChange;

      var thumbNode = /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index2["default"]
      }, function (styles) {
        return /*#__PURE__*/_react["default"].createElement(_Checkbox["default"], {
          ref: function ref(_ref) {
            return _this2.checkbox = _ref;
          },
          style: [styles.checkboxItemCheckbox, checkboxStyle],
          defaultChecked: defaultChecked,
          checked: checked,
          onChange: onChange,
          disabled: disabled
        });
      });

      return /*#__PURE__*/_react["default"].createElement(ListItem, {
        style: style,
        onPress: disabled ? undefined : this.handleClick,
        extra: extra,
        thumb: thumbNode
      }, children);
    }
  }]);
  return CheckboxItem;
}(_react["default"].Component);

exports["default"] = CheckboxItem;
