// @ant-design/react-native/lib/checkbox/Checkbox.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var Checkbox = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Checkbox, _React$Component);

  var _super = (0, _createSuper2["default"])(Checkbox);

  function Checkbox(props, context) {
    var _this;

    (0, _classCallCheck2["default"])(this, Checkbox);
    _this = _super.call(this, props, context);

    _this.handleClick = function () {
      if (_this.props.disabled) {
        return;
      }

      var checked = !_this.state.checked;

      if (!(typeof _this.props.checked === 'boolean')) {
        _this.setState({
          checked: checked
        });
      }

      if (_this.props.onChange) {
        _this.props.onChange({
          target: {
            checked: checked
          }
        });
      }
    };

    _this.state = {
      checked: props.checked || props.defaultChecked || false
    };
    return _this;
  }

  (0, _createClass2["default"])(Checkbox, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (typeof nextProps.checked === 'boolean') {
        this.setState({
          checked: !!nextProps.checked
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          disabled = _this$props.disabled,
          children = _this$props.children;
      var checked = this.state.checked;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        themeStyles: _index["default"],
        styles: this.props.styles
      }, function (styles, theme) {
        var icon;

        if (checked) {
          icon = disabled ? /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "check-square",
            style: [styles.icon, style]
          }) : /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "check-square",
            color: theme.brand_primary,
            style: [styles.icon, style]
          });
        } else {
          icon = disabled ? /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "border",
            style: [styles.icon, style]
          }) : /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "border",
            color: theme.brand_primary,
            style: [styles.icon, style]
          });
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
          onPress: _this2.handleClick
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.wrapper]
        }, icon, typeof children === 'string' ?
        /*#__PURE__*/
        // tslint:disable-next-line:jsx-no-multiline-js
        _react["default"].createElement(_reactNative.Text, {
          style: styles.iconRight
        }, _this2.props.children) : children));
      });
    }
  }]);
  return Checkbox;
}(_react["default"].Component);

exports["default"] = Checkbox;
