// @ant-design/react-native/lib/result/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _index = _interopRequireDefault(require("../button/index"));

var _style = require("../style");

var _index2 = _interopRequireDefault(require("./style/index"));

var Result = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Result, _React$Component);

  var _super = (0, _createSuper2["default"])(Result);

  function Result() {
    (0, _classCallCheck2["default"])(this, Result);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(Result, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          style = _this$props.style,
          img = _this$props.img,
          imgUrl = _this$props.imgUrl,
          title = _this$props.title,
          message = _this$props.message,
          buttonText = _this$props.buttonText,
          onButtonClick = _this$props.onButtonClick,
          buttonType = _this$props.buttonType;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index2["default"]
      }, function (styles) {
        var imgContent = null;

        if (img) {
          imgContent = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.imgWrap
          }, img);
        } else if (imgUrl) {
          imgContent = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.imgWrap
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.Image, {
            source: imgUrl,
            style: styles.img
          }));
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.result, style]
        }, imgContent, title ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.title
        }, typeof title === 'string' ? /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: styles.titleText
        }, title) : title) : null, message ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.message
        }, typeof message === 'string' ? /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: styles.messageText
        }, message) : message) : null, buttonText ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.buttonWrap
        }, /*#__PURE__*/_react["default"].createElement(_index["default"], {
          style: styles.button,
          type: buttonType,
          onPress: onButtonClick
        }, buttonText)) : null);
      });
    }
  }]);
  return Result;
}(_react["default"].Component);

exports["default"] = Result;
