// @ant-design/react-native/es/notice-bar/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { Image, Text, View } from 'react-native';
import { NoticeBar, WhiteSpace } from '../../';

var NoticeBarExample = /*#__PURE__*/function (_React$Component) {
  _inherits(NoticeBarExample, _React$Component);

  var _super = _createSuper(NoticeBarExample);

  function NoticeBarExample() {
    _classCallCheck(this, NoticeBarExample);

    return _super.apply(this, arguments);
  }

  _createClass(NoticeBarExample, [{
    key: "render",
    value: function render() {
      var customIcon = /*#__PURE__*/React.createElement(Image // tslint:disable-next-line:jsx-no-multiline-js
      , {
        // tslint:disable-next-line:jsx-no-multiline-js
        source: {
          uri: 'https://zos.alipayobjects.com/rmsportal/bRnouywfdRsCcLU.png'
        },
        style: {
          width: 12,
          height: 12
        }
      });
      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 10
        }
      }, /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        onPress: function onPress() {
          return alert('click');
        },
        marqueeProps: {
          loop: true,
          style: {
            fontSize: 12,
            color: 'red'
          }
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "closable",
        onPress: function onPress() {
          return alert('will close');
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "closable",
        icon: customIcon
      }, "Customized icon."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "link",
        onPress: function onPress() {
          return alert('link');
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "link"
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "closable",
        icon: undefined
      }, "Remove the default icon."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "closable",
        action: /*#__PURE__*/React.createElement(Text, {
          style: {
            color: '#a1a1a1'
          }
        }, "\u4E0D\u518D\u63D0\u793A")
      }, "Closable demo for `actionText`."), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(NoticeBar, {
        mode: "link",
        action: /*#__PURE__*/React.createElement(Text, null, "\u53BB\u770B\u770B")
      }, "Link demo for `actionText`."));
    }
  }]);

  return NoticeBarExample;
}(React.Component);

export { NoticeBarExample as default };
