// @ant-design/react-native/lib/progress/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var BasicProgressExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicProgressExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicProgressExample);

  function BasicProgressExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicProgressExample);
    _this = _super.call(this, props);

    _this.onAdd = function () {
      var p = _this.state.percent + 10;

      if (_this.state.percent >= 100) {
        p = 0;
      }

      _this.setState({
        percent: p
      });
    };

    _this.state = {
      percent: 40
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicProgressExample, [{
    key: "render",
    value: function render() {
      var style = {
        marginTop: 80,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
      };
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.Progress, {
        percent: 90,
        position: "fixed"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [style]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginRight: 10,
          height: 4,
          flex: 1
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Progress, {
        percent: this.state.percent
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, this.state.percent, "%")), /*#__PURE__*/_react["default"].createElement(_.Button, {
        style: {
          width: 50,
          marginLeft: 10
        },
        type: "ghost",
        size: "small",
        onPress: this.onAdd
      }, "(+-)10"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Progress, {
        percent: 5
      }));
    }
  }]);
  return BasicProgressExample;
}(_react["default"].Component);

exports["default"] = BasicProgressExample;
