// @ant-design/react-native/lib/white-space/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var PlaceHolder = function PlaceHolder(props) {
  return /*#__PURE__*/_react["default"].createElement(_reactNative.View, (0, _extends2["default"])({
    style: {
      backgroundColor: '#fff',
      height: 30,
      alignItems: 'center',
      justifyContent: 'center'
    }
  }, props), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
    style: {
      color: '#bbb'
    }
  }, "Block"));
};

var WhiteSpaceExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(WhiteSpaceExample, _React$Component);

  var _super = (0, _createSuper2["default"])(WhiteSpaceExample);

  function WhiteSpaceExample() {
    (0, _classCallCheck2["default"])(this, WhiteSpaceExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(WhiteSpaceExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xs"
      }), /*#__PURE__*/_react["default"].createElement(PlaceHolder, null), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "sm"
      }), /*#__PURE__*/_react["default"].createElement(PlaceHolder, null), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(PlaceHolder, null), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(PlaceHolder, null), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "xl"
      }), /*#__PURE__*/_react["default"].createElement(PlaceHolder, null));
    }
  }]);
  return WhiteSpaceExample;
}(_react["default"].Component);

exports["default"] = WhiteSpaceExample;
