// @ant-design/react-native/lib/notice-bar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

var _Marquee = _interopRequireDefault(require("./Marquee"));

var _index = _interopRequireDefault(require("./style/index"));

var NoticeBar = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(NoticeBar, _React$Component);

  var _super = (0, _createSuper2["default"])(NoticeBar);

  function NoticeBar(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, NoticeBar);
    _this = _super.call(this, props);

    _this.onPress = function () {
      var _this$props = _this.props,
          mode = _this$props.mode,
          onPress = _this$props.onPress;

      if (onPress) {
        onPress();
      }

      if (mode === 'closable') {
        _this.setState({
          show: false
        });
      }
    };

    _this.state = {
      show: true
    };
    return _this;
  }

  (0, _createClass2["default"])(NoticeBar, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props2 = this.props,
          children = _this$props2.children,
          mode = _this$props2.mode,
          icon = _this$props2.icon,
          style = _this$props2.style,
          action = _this$props2.action,
          marqueeProps = _this$props2.marqueeProps;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles, theme) {
        var operationDom = null;
        icon = typeof icon === 'undefined' ? /*#__PURE__*/_react["default"].createElement(_icon["default"], {
          name: "sound",
          color: theme.brand_warning
        }) : icon;

        if (mode === 'closable') {
          operationDom = /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
            onPress: _this2.onPress
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.actionWrap
          }, action ? action : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.close]
          }, "\xD7")));
        } else if (mode === 'link') {
          operationDom = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.actionWrap
          }, action ? action : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.link]
          }, "\u221F"));
        }

        var main = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.notice, style]
        }, icon && /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.left15
        }, icon), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.container, icon ? styles.left6 : styles.left15]
        }, /*#__PURE__*/_react["default"].createElement(_Marquee["default"], (0, _extends2["default"])({
          style: styles.content,
          text: children
        }, marqueeProps))), operationDom);

        return _this2.state.show ? mode === 'closable' ? main : /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
          onPress: _this2.onPress
        }, main) : null;
      });
    }
  }]);
  return NoticeBar;
}(_react["default"].Component);

exports["default"] = NoticeBar;
NoticeBar.defaultProps = {
  mode: '',
  onPress: function onPress() {}
};
