// @ant-design/react-native/lib/card/CardHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _style2 = _interopRequireDefault(require("./style"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var CardHeader = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(CardHeader, _React$Component);

  var _super = (0, _createSuper2["default"])(CardHeader);

  function CardHeader() {
    (0, _classCallCheck2["default"])(this, CardHeader);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(CardHeader, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          title = _a.title,
          thumb = _a.thumb,
          thumbStyle = _a.thumbStyle,
          extra = _a.extra,
          style = _a.style,
          styles = _a.styles,
          restProps = __rest(_a, ["title", "thumb", "thumbStyle", "extra", "style", "styles"]);

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: styles,
        themeStyles: _style2["default"]
      }, function (s) {
        var titleDom = title === undefined ? null : /*#__PURE__*/_react["default"].isValidElement(title) ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: s.headerContentWrap
        }, title) : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: s.headerContent
        }, title);
        var extraDom = extra === undefined ? null : /*#__PURE__*/_react["default"].isValidElement(extra) ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: s.headerExtraWrap
        }, extra) : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [s.headerExtra]
        }, extra);
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, (0, _extends2["default"])({
          style: [s.headerWrap, style]
        }, restProps), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [s.headerTitle]
        }, typeof thumb === 'string' ? /*#__PURE__*/_react["default"].createElement(_reactNative.Image, {
          source: {
            uri: thumb
          },
          style: [s.headerImage, thumbStyle]
        }) : thumb, titleDom), extra ? extraDom : null);
      });
    }
  }]);
  return CardHeader;
}(_react["default"].Component);

exports["default"] = CardHeader;
CardHeader.defaultProps = {
  thumbStyle: {},
  style: {}
};
