// @ant-design/react-native/lib/card/CardFooter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _style2 = _interopRequireDefault(require("./style"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var CardFooter = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(CardFooter, _React$Component);

  var _super = (0, _createSuper2["default"])(CardFooter);

  function CardFooter() {
    (0, _classCallCheck2["default"])(this, CardFooter);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(CardFooter, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          content = _a.content,
          extra = _a.extra,
          styles = _a.styles,
          style = _a.style,
          restProps = __rest(_a, ["content", "extra", "styles", "style"]);

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: styles,
        themeStyles: _style2["default"]
      }, function (s) {
        var contentDom = content !== undefined && /*#__PURE__*/_react["default"].isValidElement(content) ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: {
            flex: 1
          }
        }, content) : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: s.footerContent
        }, content);
        var extraDom = extra !== undefined && /*#__PURE__*/_react["default"].isValidElement(extra) ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: {
            flex: 1
          }
        }, extra) : /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [s.footerExtra]
        }, extra);
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, (0, _extends2["default"])({
          style: [s.footerWrap, style]
        }, restProps), contentDom, extra ? extraDom : null);
      });
    }
  }]);
  return CardFooter;
}(_react["default"].Component);

exports["default"] = CardFooter;
CardFooter.defaultProps = {
  style: {}
};
