// @ant-design/react-native/es/accordion/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { View } from 'react-native';
import { Accordion, List } from '../../';

var AccordionExmple = /*#__PURE__*/function (_React$Component) {
  _inherits(AccordionExmple, _React$Component);

  var _super = _createSuper(AccordionExmple);

  function AccordionExmple() {
    var _this;

    _classCallCheck(this, AccordionExmple);

    _this = _super.apply(this, arguments);
    _this.state = {
      activeSections: [2, 0]
    };

    _this.onChange = function (activeSections) {
      _this.setState({
        activeSections: activeSections
      });
    };

    return _this;
  }

  _createClass(AccordionExmple, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 80,
          marginBottom: 10
        }
      }, /*#__PURE__*/React.createElement(Accordion, {
        onChange: this.onChange,
        activeSections: this.state.activeSections
      }, /*#__PURE__*/React.createElement(Accordion.Panel, {
        header: "Title 1"
      }, /*#__PURE__*/React.createElement(List, null, /*#__PURE__*/React.createElement(List.Item, null, "Content 1"), /*#__PURE__*/React.createElement(List.Item, null, "Content 2"), /*#__PURE__*/React.createElement(List.Item, null, "Content 3"))), /*#__PURE__*/React.createElement(Accordion.Panel, {
        header: "Title 2"
      }, "this is panel content2 or other"), /*#__PURE__*/React.createElement(Accordion.Panel, {
        header: "Title 3"
      }, "Text text text text text text text text text text text text text text text")));
    }
  }]);

  return AccordionExmple;
}(React.Component);

export { AccordionExmple as default };
