// @ant-design/react-native/lib/drawer/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _2 = require("../../");

var styles = _reactNative.StyleSheet.create({
  container: {
    flex: 1
  }
});

var DrawerExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(DrawerExample, _React$Component);

  var _super = (0, _createSuper2["default"])(DrawerExample);

  function DrawerExample() {
    var _this;

    (0, _classCallCheck2["default"])(this, DrawerExample);
    _this = _super.apply(this, arguments);

    _this.onOpenChange = function (isOpen) {
      /* tslint:disable: no-console */
      console.log('是否打开了 Drawer', isOpen.toString());
    };

    return _this;
  }

  (0, _createClass2["default"])(DrawerExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var itemArr = Array.apply(null, Array(20)).map(function (_, i) {
        return i;
      }).map(function (_i, index) {
        if (index === 0) {
          return /*#__PURE__*/_react["default"].createElement(_2.List.Item, {
            key: index,
            thumb: "https://zos.alipayobjects.com/rmsportal/eOZidTabPoEbPeU.png",
            multipleLine: true
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: {
              flexDirection: 'row',
              justifyContent: 'space-between',
              alignItems: 'center'
            }
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Categories - ", index), /*#__PURE__*/_react["default"].createElement(_2.Button, {
            type: "primary",
            size: "small",
            onPress: function onPress() {
              return _this2.drawer.closeDrawer();
            }
          }, "close drawer")));
        }

        return /*#__PURE__*/_react["default"].createElement(_2.List.Item, {
          key: index,
          thumb: "https://zos.alipayobjects.com/rmsportal/eOZidTabPoEbPeU.png"
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Categories - ", index));
      }); // Todo: https://github.com/DefinitelyTyped/DefinitelyTyped

      var sidebar = /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
        style: [styles.container]
      }, /*#__PURE__*/_react["default"].createElement(_2.List, null, itemArr));

      return /*#__PURE__*/_react["default"].createElement(_2.Drawer, {
        sidebar: sidebar,
        position: "left",
        open: false,
        drawerRef: function drawerRef(el) {
          return _this2.drawer = el;
        },
        onOpenChange: this.onOpenChange,
        drawerBackgroundColor: "#ccc"
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          flex: 1,
          marginTop: 114,
          padding: 8
        }
      }, /*#__PURE__*/_react["default"].createElement(_2.Button, {
        onPress: function onPress() {
          return _this2.drawer && _this2.drawer.openDrawer();
        }
      }, "Open drawer"), /*#__PURE__*/_react["default"].createElement(_2.WhiteSpace, null)));
    }
  }]);
  return DrawerExample;
}(_react["default"].Component);

exports["default"] = DrawerExample;
