// @ant-design/react-native/lib/search-bar/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var SearchBarDemo = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(SearchBarDemo, _React$Component);

  var _super = (0, _createSuper2["default"])(SearchBarDemo);

  function SearchBarDemo() {
    var _this;

    (0, _classCallCheck2["default"])(this, SearchBarDemo);
    _this = _super.apply(this, arguments);
    _this.state = {
      value: '美食'
    };

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.clear = function () {
      _this.setState({
        value: ''
      });
    };

    return _this;
  }

  (0, _createClass2["default"])(SearchBarDemo, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_.SearchBar, {
        defaultValue: "\u521D\u59CB\u503C",
        placeholder: "\u641C\u7D22"
      }), /*#__PURE__*/_react["default"].createElement(_.SearchBar, {
        value: this.state.value,
        placeholder: "\u641C\u7D22",
        onSubmit: function onSubmit(value) {
          return _reactNative.Alert.alert(value);
        },
        onCancel: this.clear,
        onChange: this.onChange,
        showCancelButton: true
      }));
    }
  }]);
  return SearchBarDemo;
}(_react["default"].Component);

exports["default"] = SearchBarDemo;
