// @ant-design/react-native/es/slider/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Slider } from '../../';

var BasicSliderExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicSliderExample, _React$Component);

  var _super = _createSuper(BasicSliderExample);

  function BasicSliderExample(props) {
    var _this;

    _classCallCheck(this, BasicSliderExample);

    _this = _super.call(this, props);

    _this.handleChange = function (value) {
      _this.setState({
        changingValue: value
      });
    };

    _this.onAfterChange = function (value) {
      _this.setState({
        changedValue: value
      });
    };

    _this.minMaxChange = function (value) {
      _this.setState({
        minMaxValue: value
      });
    };

    _this.state = {
      changingValue: 0.25,
      changedValue: 0.15,
      minMaxValue: 0,
      slideCompletionCount: 0
    };
    return _this;
  }

  _createClass(BasicSliderExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Text, null, "Default settings"), /*#__PURE__*/React.createElement(Slider, null)), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "Initial value: 0.5"), /*#__PURE__*/React.createElement(Slider, {
        defaultValue: 0.5
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "min: 0, max: 1, current Value: ", this.state.minMaxValue), /*#__PURE__*/React.createElement(Slider, {
        min: 0,
        max: 1,
        onAfterChange: function onAfterChange(value) {
          return _this2.minMaxChange(value);
        }
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "step: 0.25"), /*#__PURE__*/React.createElement(Slider, {
        step: 0.25,
        value: 0.25
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "disabled"), /*#__PURE__*/React.createElement(Slider, {
        disabled: true,
        defaultValue: 0.25
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "onChange value: ", this.state.changingValue), /*#__PURE__*/React.createElement(Slider, {
        defaultValue: 0.25,
        onChange: function onChange(value) {
          return _this2.handleChange(value);
        }
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "onAfterChange value: ", this.state.changedValue), /*#__PURE__*/React.createElement(Slider, {
        defaultValue: 0.15,
        onAfterChange: function onAfterChange(value) {
          return _this2.onAfterChange(value);
        }
      })), /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/React.createElement(Text, null, "custom color: "), /*#__PURE__*/React.createElement(Slider, {
        defaultValue: 0.15,
        minimumTrackTintColor: "red",
        maximumTrackTintColor: "blue"
      })));
    }
  }]);

  return BasicSliderExample;
}(React.Component);

export { BasicSliderExample as default };
