// @ant-design/react-native/lib/tabs/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.description = exports.title = exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = _interopRequireDefault(require(".."));

/* tslint:disable:no-console */
var renderContent = function renderContent(tab, index) {
  var style = {
    paddingVertical: 40,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    backgroundColor: '#ddd'
  };
  var content = [1, 2, 3, 4, 5, 6, 7, 8].map(function (i) {
    return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
      key: "".concat(index, "_").concat(i),
      style: style
    }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, tab.title, " - ", i));
  });
  return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
    style: {
      backgroundColor: '#fff'
    }
  }, content);
};

var BasicTabsExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicTabsExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicTabsExample);

  function BasicTabsExample() {
    (0, _classCallCheck2["default"])(this, BasicTabsExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicTabsExample, [{
    key: "render",
    value: function render() {
      var tabs = [{
        title: 'First Tab'
      }, {
        title: 'Second Tab'
      }, {
        title: 'Third Tab'
      }];
      var tabs2 = [{
        title: '1st Tab'
      }, {
        title: '2nd Tab'
      }, {
        title: '3rd Tab'
      }, {
        title: '4th Tab'
      }, {
        title: '5th Tab'
      }, {
        title: '6th Tab'
      }, {
        title: '7th Tab'
      }, {
        title: '8th Tab'
      }, {
        title: '9th Tab'
      }];
      var style = {
        alignItems: 'center',
        justifyContent: 'center',
        height: 150,
        backgroundColor: '#fff'
      };
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          flex: 1
        }
      }, /*#__PURE__*/_react["default"].createElement(_["default"], {
        tabs: tabs
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of First Tab")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of Second Tab")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of Third Tab"))), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 16
        }
      }, "Custom RenderTabBar"), /*#__PURE__*/_react["default"].createElement(_["default"], {
        tabs: tabs,
        renderTabBar: function renderTabBar(tabProps) {
          return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: {
              paddingHorizontal: 16,
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'space-evenly'
            }
          }, tabProps.tabs.map(function (tab, i) {
            return (
              /*#__PURE__*/
              // change the style to fit your needs
              _react["default"].createElement(_reactNative.TouchableOpacity, {
                activeOpacity: 0.9,
                key: tab.key || i,
                style: {
                  // width: '30%',
                  padding: 6
                },
                onPress: function onPress() {
                  var goToTab = tabProps.goToTab,
                      onTabClick = tabProps.onTabClick; // tslint:disable-next-line:no-unused-expression

                  onTabClick && onTabClick(tabs[i], i); // tslint:disable-next-line:no-unused-expression

                  goToTab && goToTab(i);
                }
              }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
                style: {
                  color: tabProps.activeTab === i ? 'green' : undefined
                }
              }, tab.title))
            );
          }));
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of First Tab")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of Second Tab")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: style
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Content of Third Tab"))), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          flex: 2
        }
      }, /*#__PURE__*/_react["default"].createElement(_["default"], {
        tabs: tabs2,
        initialPage: 1,
        tabBarPosition: "top"
      }, renderContent)));
    }
  }]);
  return BasicTabsExample;
}(_react["default"].Component);

exports["default"] = BasicTabsExample;
var title = 'Tabs';
exports.title = title;
var description = 'Tabs example';
exports.description = description;
