// @ant-design/react-native/lib/checkbox/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var AgreeItem = _.Checkbox.AgreeItem;
var CheckboxItem = _.Checkbox.CheckboxItem;

var BasicCheckboxExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicCheckboxExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicCheckboxExample);

  function BasicCheckboxExample(props, context) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicCheckboxExample);
    _this = _super.call(this, props, context);
    _this.state = {
      checkBox1: true,
      agreeItem1: true,
      checkboxItem1: true
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicCheckboxExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Checkbox, {
        checked: this.state.checkBox1,
        style: {
          color: '#f00'
        },
        onChange: function onChange(event) {
          _this2.setState({
            checkBox1: event.target.checked
          });
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Checkbox, null, "Checkbox"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Checkbox, {
        checked: true,
        disabled: true
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Checkbox, {
        disabled: true
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(AgreeItem, null, "Agree agreement agreement agreement agreement agreement agreement agreement"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(AgreeItem, {
        checked: this.state.agreeItem1,
        checkboxStyle: {
          color: '#f00'
        },
        onChange: function onChange(event) {
          _this2.setState({
            agreeItem1: event.target.checked
          });
        }
      }, "Agree agreement"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(AgreeItem, {
        disabled: true
      }, "Not selected. Not editable"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(AgreeItem, {
        checked: true,
        disabled: true
      }, "Force selected. Not editable"), /*#__PURE__*/_react["default"].createElement(_.List, {
        style: {
          marginTop: 12
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          marginTop: 12
        }
      }, "Multiple options"), /*#__PURE__*/_react["default"].createElement(CheckboxItem, {
        checked: this.state.checkboxItem1,
        onChange: function onChange(event) {
          _this2.setState({
            checkboxItem1: event.target.checked
          });
        }
      }, "Option 1"), /*#__PURE__*/_react["default"].createElement(CheckboxItem, null, "Option 2"), /*#__PURE__*/_react["default"].createElement(CheckboxItem, {
        disabled: true
      }, "Option 3"), /*#__PURE__*/_react["default"].createElement(CheckboxItem, {
        disabled: true,
        checked: true
      }, "Option 4")));
    }
  }]);
  return BasicCheckboxExample;
}(_react["default"].Component);

exports["default"] = BasicCheckboxExample;
