// @ant-design/react-native/lib/portal/portal-manager.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

/**
 * Portal host is the component which actually renders all Portals.
 */
var PortalManager = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2["default"])(PortalManager, _React$PureComponent);

  var _super = (0, _createSuper2["default"])(PortalManager);

  function PortalManager() {
    var _this;

    (0, _classCallCheck2["default"])(this, PortalManager);
    _this = _super.apply(this, arguments);
    _this.state = {
      portals: []
    };

    _this.mount = function (key, children) {
      _this.setState(function (state) {
        return {
          portals: [].concat((0, _toConsumableArray2["default"])(state.portals), [{
            key: key,
            children: children
          }])
        };
      });
    };

    _this.update = function (key, children) {
      return _this.setState(function (state) {
        return {
          portals: state.portals.map(function (item) {
            if (item.key === key) {
              return (0, _extends2["default"])((0, _extends2["default"])({}, item), {
                children: children
              });
            }

            return item;
          })
        };
      });
    };

    _this.unmount = function (key) {
      return _this.setState(function (state) {
        return {
          portals: state.portals.filter(function (item) {
            return item.key !== key;
          })
        };
      });
    };

    return _this;
  }

  (0, _createClass2["default"])(PortalManager, [{
    key: "render",
    value: function render() {
      return this.state.portals.map(function (_ref, i) {
        var key = _ref.key,
            children = _ref.children;
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          key: key,
          collapsable: false
          /* Need collapsable=false here to clip the elevations, otherwise they appear above sibling components */
          ,
          pointerEvents: "box-none",
          style: [_reactNative.StyleSheet.absoluteFill, {
            zIndex: 1000 + i
          }]
        }, children);
      });
    }
  }]);
  return PortalManager;
}(_react["default"].PureComponent);

exports["default"] = PortalManager;
