// @ant-design/react-native/es/card/CardFooter.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var __rest = this && this.__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

import React from 'react';
import { Text, View } from 'react-native';
import { WithTheme } from '../style';
import CardStyles from './style';

var CardFooter = /*#__PURE__*/function (_React$Component) {
  _inherits(CardFooter, _React$Component);

  var _super = _createSuper(CardFooter);

  function CardFooter() {
    _classCallCheck(this, CardFooter);

    return _super.apply(this, arguments);
  }

  _createClass(CardFooter, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          content = _a.content,
          extra = _a.extra,
          styles = _a.styles,
          style = _a.style,
          restProps = __rest(_a, ["content", "extra", "styles", "style"]);

      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: styles,
        themeStyles: CardStyles
      }, function (s) {
        var contentDom = content !== undefined && /*#__PURE__*/React.isValidElement(content) ? /*#__PURE__*/React.createElement(View, {
          style: {
            flex: 1
          }
        }, content) : /*#__PURE__*/React.createElement(Text, {
          style: s.footerContent
        }, content);
        var extraDom = extra !== undefined && /*#__PURE__*/React.isValidElement(extra) ? /*#__PURE__*/React.createElement(View, {
          style: {
            flex: 1
          }
        }, extra) : /*#__PURE__*/React.createElement(Text, {
          style: [s.footerExtra]
        }, extra);
        return /*#__PURE__*/React.createElement(View, _extends({
          style: [s.footerWrap, style]
        }, restProps), contentDom, extra ? extraDom : null);
      });
    }
  }]);

  return CardFooter;
}(React.Component);

export { CardFooter as default };
CardFooter.defaultProps = {
  style: {}
};
