// @ant-design/react-native/es/card/CardHeader.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var __rest = this && this.__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

import React from 'react';
import { Image, Text, View } from 'react-native';
import { WithTheme } from '../style';
import CardStyles from './style';

var CardHeader = /*#__PURE__*/function (_React$Component) {
  _inherits(CardHeader, _React$Component);

  var _super = _createSuper(CardHeader);

  function CardHeader() {
    _classCallCheck(this, CardHeader);

    return _super.apply(this, arguments);
  }

  _createClass(CardHeader, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          title = _a.title,
          thumb = _a.thumb,
          thumbStyle = _a.thumbStyle,
          extra = _a.extra,
          style = _a.style,
          styles = _a.styles,
          restProps = __rest(_a, ["title", "thumb", "thumbStyle", "extra", "style", "styles"]);

      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: styles,
        themeStyles: CardStyles
      }, function (s) {
        var titleDom = title === undefined ? null : /*#__PURE__*/React.isValidElement(title) ? /*#__PURE__*/React.createElement(View, {
          style: s.headerContentWrap
        }, title) : /*#__PURE__*/React.createElement(Text, {
          style: s.headerContent
        }, title);
        var extraDom = extra === undefined ? null : /*#__PURE__*/React.isValidElement(extra) ? /*#__PURE__*/React.createElement(View, {
          style: s.headerExtraWrap
        }, extra) : /*#__PURE__*/React.createElement(Text, {
          style: [s.headerExtra]
        }, extra);
        return /*#__PURE__*/React.createElement(View, _extends({
          style: [s.headerWrap, style]
        }, restProps), /*#__PURE__*/React.createElement(View, {
          style: [s.headerTitle]
        }, typeof thumb === 'string' ? /*#__PURE__*/React.createElement(Image, {
          source: {
            uri: thumb
          },
          style: [s.headerImage, thumbStyle]
        }) : thumb, titleDom), extra ? extraDom : null);
      });
    }
  }]);

  return CardHeader;
}(React.Component);

export { CardHeader as default };
CardHeader.defaultProps = {
  thumbStyle: {},
  style: {}
};
