// @ant-design/react-native/es/search-bar/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Alert, View } from 'react-native';
import { SearchBar } from '../../';

var SearchBarDemo = /*#__PURE__*/function (_React$Component) {
  _inherits(SearchBarDemo, _React$Component);

  var _super = _createSuper(SearchBarDemo);

  function SearchBarDemo() {
    var _this;

    _classCallCheck(this, SearchBarDemo);

    _this = _super.apply(this, arguments);
    _this.state = {
      value: '美食'
    };

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.clear = function () {
      _this.setState({
        value: ''
      });
    };

    return _this;
  }

  _createClass(SearchBarDemo, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/React.createElement(SearchBar, {
        defaultValue: "\u521D\u59CB\u503C",
        placeholder: "\u641C\u7D22"
      }), /*#__PURE__*/React.createElement(SearchBar, {
        value: this.state.value,
        placeholder: "\u641C\u7D22",
        onSubmit: function onSubmit(value) {
          return Alert.alert(value);
        },
        onCancel: this.clear,
        onChange: this.onChange,
        showCancelButton: true
      }));
    }
  }]);

  return SearchBarDemo;
}(React.Component);

export { SearchBarDemo as default };
