// @ant-design/react-native/lib/steps/StepsItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

/* tslint:disable: jsx-no-multiline-js */
var StepsItem = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(StepsItem, _React$Component);

  var _super = (0, _createSuper2["default"])(StepsItem);

  function StepsItem() {
    (0, _classCallCheck2["default"])(this, StepsItem);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(StepsItem, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          last = _this$props.last,
          title = _this$props.title,
          description = _this$props.description,
          direction = _this$props.direction,
          status = _this$props.status,
          icon = _this$props.icon,
          styles = _this$props.styles,
          renderIcon = _this$props.renderIcon;
      var index = this.props.index;
      var current = this.props.current;
      var errorTail = this.props.errorTail;
      var starting = index < current || status === 'finish' || index === current || status === 'process';
      var waiting = index > current || status === 'wait';
      var error = status === 'error';
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, null, function (_, theme) {
        var headCls = '';
        var tailTopCls = '';
        var tailBottomCls = '';
        var sizeCls = size === 'small' ? '_s' : '_l';

        if (index < current || status === 'finish') {
          headCls = "head_blue".concat(sizeCls);
          tailTopCls = 'tail_blue';
          tailBottomCls = 'tail_blue';
        } else if (index === current || status === 'process') {
          headCls = "head_blue".concat(sizeCls);
          tailTopCls = 'tail_blue';
          tailBottomCls = 'tail_gray';
        } else if (index > current || status === 'wait') {
          headCls = "head_gray".concat(sizeCls);
          tailTopCls = 'tail_gray';
          tailBottomCls = 'tail_gray';
        } else if (status === 'error') {
          headCls = "head_red".concat(sizeCls);
          tailTopCls = 'tail_gray';
          tailBottomCls = 'tail_gray';
        }

        if (last) {
          tailTopCls = 'tail_last';
          tailBottomCls = 'tail_last';
        }

        if (errorTail > -1) {
          tailBottomCls = 'tail_error';
        }

        var iconSource;

        if (renderIcon) {
          iconSource = renderIcon({
            starting: starting,
            waiting: waiting,
            error: error
          });
        } else {
          if (starting) {
            iconSource = /*#__PURE__*/_react["default"].createElement(_icon["default"], {
              name: "check",
              color: theme.color_icon_base,
              style: styles["icon".concat(sizeCls)]
            });
          } else if (waiting) {
            iconSource = /*#__PURE__*/_react["default"].createElement(_icon["default"], {
              name: "ellipsis",
              color: theme.color_icon_base,
              style: styles["icon".concat(sizeCls)]
            });

            if (icon) {
              iconSource = icon;
            }
          } else if (error) {
            iconSource = /*#__PURE__*/_react["default"].createElement(_icon["default"], {
              name: "close",
              color: theme.color_icon_base,
              style: styles["icon".concat(sizeCls)]
            });
          }
        }

        var isHorizontal = direction === 'horizontal';
        var parentStyle = isHorizontal ? {
          flexDirection: 'column'
        } : {
          flexDirection: 'row'
        };
        var childStyle = isHorizontal ? {
          flexDirection: 'row',
          flex: 1
        } : {
          flexDirection: 'column'
        };
        var styleSuffix = '';

        if (isHorizontal) {
          styleSuffix = '_h';
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: parentStyle
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: childStyle
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles["head_default".concat(sizeCls)], styles[headCls]]
        }, /*#__PURE__*/_react["default"].isValidElement(iconSource) ? iconSource : null), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles["tail_default".concat(sizeCls).concat(styleSuffix)], styles[tailTopCls]]
        }), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles["tail_default".concat(sizeCls).concat(styleSuffix)], styles[tailBottomCls]]
        })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles["content".concat(sizeCls).concat(styleSuffix)]
        }, (0, _typeof2["default"])(title) !== 'object' ? /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles["title".concat(sizeCls)]]
        }, title) : /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, title), (0, _typeof2["default"])(description) !== 'object' ? /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles["description".concat(sizeCls)]]
        }, description) : /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, description)));
      });
    }
  }]);
  return StepsItem;
}(_react["default"].Component);

exports["default"] = StepsItem;
