// @ant-design/react-native/lib/progress/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var Progress = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Progress, _React$Component);

  var _super = (0, _createSuper2["default"])(Progress);

  function Progress(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Progress);
    _this = _super.call(this, props);

    _this.onLayout = function (e) {
      _this.setState({
        wrapWidth: e.nativeEvent.layout.width
      });
    };

    _this.normalPercent = function (percent) {
      var widthPercent = 0;

      if (percent !== undefined && percent > 0) {
        widthPercent = percent > 100 ? 100 : percent;
      }

      return widthPercent;
    };

    _this.getWidth = function () {
      var percent = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _this.props.percent;
      return _this.state.wrapWidth * (_this.normalPercent(percent) / 100);
    };

    _this.state = {
      wrapWidth: props.wrapWidth || _reactNative.Dimensions.get('window').width,
      percentage: new _reactNative.Animated.Value(0)
    };
    return _this;
  }

  (0, _createClass2["default"])(Progress, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (nextProps.wrapWidth !== this.props.wrapWidth) {
        this.setState({
          wrapWidth: nextProps.wrapWidth
        });
      }

      if (this.props.appearTransition && nextProps.percent !== this.props.percent) {
        this.setState({
          percentage: new _reactNative.Animated.Value(this.getWidth(nextProps.percent))
        });
      }
    }
  }, {
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.props.appearTransition) {
        this.state.percentage.setValue(0);

        _reactNative.Animated.timing(this.state.percentage, {
          toValue: this.getWidth(),
          duration: 1000,
          useNativeDriver: true
        }).start();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          position = _this$props.position,
          unfilled = _this$props.unfilled,
          style = _this$props.style,
          barStyle = _this$props.barStyle;
      var percentStyle = {
        width: this.getWidth(),
        height: 0
      };
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        var child = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.progressBar, percentStyle, barStyle]
        });

        if (_this2.props.appearTransition) {
          percentStyle.width = _this2.state.percentage;
          child = /*#__PURE__*/_react["default"].createElement(_reactNative.Animated.View, {
            style: [styles.progressBar, percentStyle, barStyle]
          });
        }

        var outerStyle = [styles.progressOuter, position === 'fixed' ? {
          position: 'absolute',
          top: 0
        } : {}, !unfilled ? {
          backgroundColor: 'transparent'
        } : {}, style];
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          onLayout: _this2.onLayout,
          style: outerStyle
        }, child);
      });
    }
  }]);
  return Progress;
}(_react["default"].Component);

exports["default"] = Progress;
Progress.defaultProps = {
  percent: 0,
  position: 'normal',
  unfilled: true,
  appearTransition: false
};
