// @ant-design/react-native/lib/segmented-control/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var BasicTagExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicTagExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicTagExample);

  function BasicTagExample() {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicTagExample);
    _this = _super.apply(this, arguments);

    _this.onChange = function (e) {
      console.log("selectedIndex:".concat(e.nativeEvent.selectedSegmentIndex));
    };

    _this.onValueChange = function (value) {
      console.log(value);
    };

    return _this;
  }

  (0, _createClass2["default"])(BasicTagExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingTop: 60,
          paddingHorizontal: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Disabled"), /*#__PURE__*/_react["default"].createElement(_.SegmentedControl, {
        values: ['Segment1', 'Segment2'],
        disabled: true
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "TintColor and Style"), /*#__PURE__*/_react["default"].createElement(_.SegmentedControl, {
        values: ['Segment1', 'Segment2', 'Segment3'],
        tintColor: '#ff0000',
        style: {
          height: 40,
          width: 280
        }
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "SelectedIndex"), /*#__PURE__*/_react["default"].createElement(_.SegmentedControl, {
        selectedIndex: 1,
        values: ['Segment1', 'Segment2', 'Segment3']
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "onChange/onValueChange"), /*#__PURE__*/_react["default"].createElement(_.SegmentedControl, {
        values: ['Segment1', 'Segment2', 'Segment3'],
        onChange: this.onChange,
        onValueChange: this.onValueChange
      }));
    }
  }]);
  return BasicTagExample;
}(_react["default"].Component);

exports["default"] = BasicTagExample;
