// @ant-design/react-native/es/portal/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Button, Icon, Portal } from '../..';

var PopoverExample = /*#__PURE__*/function (_React$Component) {
  _inherits(PopoverExample, _React$Component);

  var _super = _createSuper(PopoverExample);

  function PopoverExample() {
    _classCallCheck(this, PopoverExample);

    return _super.apply(this, arguments);
  }

  _createClass(PopoverExample, [{
    key: "render",
    value: function render() {
      var key;
      var contents = /*#__PURE__*/React.createElement(View, {
        style: {
          backgroundColor: 'green',
          padding: 100
        }
      }, /*#__PURE__*/React.createElement(Button, {
        onPress: function onPress() {
          Portal.remove(key);
        },
        type: "primary",
        style: {
          position: 'absolute',
          top: 100,
          right: 0
        }
      }, /*#__PURE__*/React.createElement(Icon, {
        name: "close-circle"
      })), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Text, null, "\u81EA\u5B9A\u4E49Portal")));
      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Button, {
        onPress: function onPress() {
          key = Portal.add(contents);
        }
      }, "Open Portal"));
    }
  }]);

  return PopoverExample;
}(React.Component);

export { PopoverExample as default };
export var title = 'Portal';
export var description = 'Portal example';
