// @ant-design/react-native/es/drawer/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import { Button, Drawer, List, WhiteSpace } from '../../';
var styles = StyleSheet.create({
  container: {
    flex: 1
  }
});

var DrawerExample = /*#__PURE__*/function (_React$Component) {
  _inherits(DrawerExample, _React$Component);

  var _super = _createSuper(DrawerExample);

  function DrawerExample() {
    var _this;

    _classCallCheck(this, DrawerExample);

    _this = _super.apply(this, arguments);

    _this.onOpenChange = function (isOpen) {
      /* tslint:disable: no-console */
      console.log('是否打开了 Drawer', isOpen.toString());
    };

    return _this;
  }

  _createClass(DrawerExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var itemArr = Array.apply(null, Array(20)).map(function (_, i) {
        return i;
      }).map(function (_i, index) {
        if (index === 0) {
          return /*#__PURE__*/React.createElement(List.Item, {
            key: index,
            thumb: "https://zos.alipayobjects.com/rmsportal/eOZidTabPoEbPeU.png",
            multipleLine: true
          }, /*#__PURE__*/React.createElement(View, {
            style: {
              flexDirection: 'row',
              justifyContent: 'space-between',
              alignItems: 'center'
            }
          }, /*#__PURE__*/React.createElement(Text, null, "Categories - ", index), /*#__PURE__*/React.createElement(Button, {
            type: "primary",
            size: "small",
            onPress: function onPress() {
              return _this2.drawer.closeDrawer();
            }
          }, "close drawer")));
        }

        return /*#__PURE__*/React.createElement(List.Item, {
          key: index,
          thumb: "https://zos.alipayobjects.com/rmsportal/eOZidTabPoEbPeU.png"
        }, /*#__PURE__*/React.createElement(Text, null, "Categories - ", index));
      }); // Todo: https://github.com/DefinitelyTyped/DefinitelyTyped

      var sidebar = /*#__PURE__*/React.createElement(ScrollView, {
        style: [styles.container]
      }, /*#__PURE__*/React.createElement(List, null, itemArr));
      return /*#__PURE__*/React.createElement(Drawer, {
        sidebar: sidebar,
        position: "left",
        open: false,
        drawerRef: function drawerRef(el) {
          return _this2.drawer = el;
        },
        onOpenChange: this.onOpenChange,
        drawerBackgroundColor: "#ccc"
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          flex: 1,
          marginTop: 114,
          padding: 8
        }
      }, /*#__PURE__*/React.createElement(Button, {
        onPress: function onPress() {
          return _this2.drawer && _this2.drawer.openDrawer();
        }
      }, "Open drawer"), /*#__PURE__*/React.createElement(WhiteSpace, null)));
    }
  }]);

  return DrawerExample;
}(React.Component);

export { DrawerExample as default };
