// @ant-design/react-native/lib/modal/Modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _getLocale = require("../_util/getLocale");

var _zh_CN = _interopRequireDefault(require("./locale/zh_CN"));

var _ModalView = _interopRequireDefault(require("./ModalView"));

var _index = _interopRequireDefault(require("./style/index"));

var _localeProvider = require("../locale-provider");

var AntmModal = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(AntmModal, _React$Component);

  var _super = (0, _createSuper2["default"])(AntmModal);

  function AntmModal() {
    (0, _classCallCheck2["default"])(this, AntmModal);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(AntmModal, [{
    key: "render",
    value: function render() {
      var _this = this;

      var _this$props = this.props,
          title = _this$props.title,
          closable = _this$props.closable,
          footer = _this$props.footer,
          children = _this$props.children,
          style = _this$props.style,
          animateAppear = _this$props.animateAppear,
          maskClosable = _this$props.maskClosable,
          popup = _this$props.popup,
          transparent = _this$props.transparent,
          visible = _this$props.visible,
          onClose = _this$props.onClose,
          bodyStyle = _this$props.bodyStyle,
          onAnimationEnd = _this$props.onAnimationEnd,
          onRequestClose = _this$props.onRequestClose; // tslint:disable-next-line:variable-name

      var _locale = (0, _getLocale.getComponentLocale)(this.props, this.context, 'Modal', function () {
        return _zh_CN["default"];
      });

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        var btnGroupStyle = styles.buttonGroupV;
        var horizontalFlex = {};

        if (footer && footer.length === 2 && !_this.props.operation) {
          btnGroupStyle = styles.buttonGroupH;
          horizontalFlex = {
            flex: 1
          };
        }

        var buttonWrapStyle = footer && footer.length === 2 ? styles.buttonWrapH : styles.buttonWrapV;
        var footerDom;

        if (footer && footer.length) {
          var footerButtons = footer.map(function (button, i) {
            var buttonStyle = {};

            if (_this.props.operation) {
              buttonStyle = styles.buttonTextOperation;
            }

            if (button.style) {
              buttonStyle = button.style;

              if (typeof buttonStyle === 'string') {
                var styleMap = {
                  cancel: {},
                  "default": {},
                  destructive: {
                    color: 'red'
                  }
                };
                buttonStyle = styleMap[buttonStyle] || {};
              }
            }

            var noneBorder = footer && footer.length === 2 && i === 1 ? {
              borderRightWidth: 0
            } : {};

            var onPressFn = function onPressFn() {
              if (button.onPress) {
                button.onPress();
              }

              if (onClose) {
                onClose();
              }
            };

            return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableHighlight, {
              key: i,
              style: horizontalFlex,
              underlayColor: "#ddd",
              onPress: onPressFn
            }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
              style: [buttonWrapStyle, noneBorder]
            }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
              style: [styles.buttonText, buttonStyle]
            }, button.text || "".concat(_locale.buttonText).concat(i))));
          });
          footerDom = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [btnGroupStyle, styles.footer]
          }, footerButtons);
        }

        var animType = _this.props.animationType;

        if (transparent) {
          if (animType === 'slide') {
            animType = 'slide-up';
          }

          var closableDom = closable ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [styles.closeWrap]
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
            onPress: onClose
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.close]
          }, "\xD7")))) : null;
          return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.container
          }, /*#__PURE__*/_react["default"].createElement(_ModalView["default"], {
            onClose: onClose,
            animationType: animType,
            wrapStyle: transparent ? styles.wrap : undefined,
            style: styles.wrap,
            visible: visible,
            onAnimationEnd: onAnimationEnd,
            onRequestClose: onRequestClose,
            animateAppear: animateAppear,
            maskClosable: maskClosable
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.KeyboardAvoidingView, {
            behavior: "padding",
            enabled: _reactNative.Platform.OS === 'ios'
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [styles.innerContainer, style]
          }, title ? /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.header]
          }, title) : null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [styles.body, bodyStyle]
          }, children), footerDom, closableDom))));
        }

        if (popup) {
          var aType = 'SlideDown';

          if (animType === 'slide-up') {
            animType = 'slide-up';
            aType = 'SlideUp';
          } else {
            animType = 'slide-down';
          }

          return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: styles.container
          }, /*#__PURE__*/_react["default"].createElement(_ModalView["default"], {
            onClose: onClose,
            animationType: animType,
            // tslint:disable-next-line:jsx-no-multiline-js
            style: [styles.popupContainer, styles["popup".concat(aType)], style],
            visible: visible,
            onAnimationEnd: onAnimationEnd,
            onRequestClose: onRequestClose,
            animateAppear: animateAppear,
            maskClosable: maskClosable
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: bodyStyle
          }, children)));
        }

        if (animType === 'slide') {
          animType = undefined;
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.container
        }, /*#__PURE__*/_react["default"].createElement(_ModalView["default"], {
          visible: visible,
          animationType: animType,
          onRequestClose: onRequestClose,
          onClose: onClose
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: style
        }, children)));
      });
    }
  }]);
  return AntmModal;
}(_react["default"].Component);

AntmModal.defaultProps = {
  visible: false,
  closable: false,
  maskClosable: false,
  style: {},
  bodyStyle: {},
  animationType: 'fade',
  onClose: function onClose() {},
  footer: [],
  transparent: false,
  popup: false,
  animateAppear: true,
  operation: false
};
AntmModal.contextType = _localeProvider.LocaleContext;
var _default = AntmModal;
exports["default"] = _default;
