// @ant-design/react-native/es/badge/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { ScrollView, View } from 'react-native';
import { Badge, WhiteSpace } from '../../';

var BasicTagExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicTagExample, _React$Component);

  var _super = _createSuper(BasicTagExample);

  function BasicTagExample() {
    _classCallCheck(this, BasicTagExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicTagExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(ScrollView, {
        style: {
          flex: 1
        },
        automaticallyAdjustContentInsets: false,
        showsHorizontalScrollIndicator: false,
        showsVerticalScrollIndicator: false
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          padding: 20
        }
      }, /*#__PURE__*/React.createElement(Badge, {
        text: 9
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Badge, {
        text: 109,
        overflowCount: 100
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Badge, {
        text: 109
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Badge, {
        text: "new"
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Badge, {
        text: 109,
        dot: true
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Badge, {
        text: 33,
        corner: true
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      }))));
    }
  }]);

  return BasicTagExample;
}(React.Component);

export { BasicTagExample as default };
