// @sentry/wizard/dist/lib/Steps/Integrations/NextJs.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NextJs = void 0;
var fs = require("fs");
var inquirer_1 = require("inquirer");
var _ = require("lodash");
var path = require("path");
var Logging_1 = require("../../Helper/Logging");
var SentryCli_1 = require("../../Helper/SentryCli");
var BaseIntegration_1 = require("./BaseIntegration");
var MIN_NEXTJS_VERSION = '10.0.0';
var PROPERTIES_FILENAME = 'sentry.properties';
var CONFIG_DIR = 'configs/';
var MERGEABLE_CONFIG_PREFIX = '_';
var appPackage = {};
try {
    appPackage = require(path.join(process.cwd(), 'package.json'));
}
catch (_a) {
    // We don't need to have this
}
var NextJs = /** @class */ (function (_super) {
    __extends(NextJs, _super);
    function NextJs(_argv) {
        var _this = _super.call(this, _argv) || this;
        _this._argv = _argv;
        _this._sentryCli = new SentryCli_1.SentryCli(_this._argv);
        return _this;
    }
    NextJs.prototype.emit = function (answers) {
        return __awaiter(this, void 0, void 0, function () {
            var dsn, sentryCliProps, configDirectory;
            return __generator(this, function (_a) {
                dsn = _.get(answers, ['config', 'dsn', 'public'], null);
                Logging_1.nl();
                sentryCliProps = this._sentryCli.convertAnswersToProperties(answers);
                fs.writeFileSync("./" + PROPERTIES_FILENAME, this._sentryCli.dumpProperties(sentryCliProps));
                Logging_1.green("Successfully created sentry.properties");
                Logging_1.nl();
                configDirectory = path.join(__dirname, '..', '..', '..', 'NextJs', CONFIG_DIR);
                if (fs.existsSync(configDirectory)) {
                    this._createNextConfig(configDirectory, dsn);
                }
                else {
                    Logging_1.debug("Couldn't find " + configDirectory + ", probably because you run from src");
                    Logging_1.nl();
                }
                Logging_1.l('For more information, see https://docs.sentry.io/platforms/javascript/guides/nextjs/');
                Logging_1.nl();
                return [2 /*return*/, {}];
            });
        });
    };
    NextJs.prototype.shouldConfigure = function (_answers) {
        return __awaiter(this, void 0, void 0, function () {
            var userAnswers;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (this._shouldConfigure) {
                            return [2 /*return*/, this._shouldConfigure];
                        }
                        Logging_1.nl();
                        userAnswers = { continue: true };
                        if (!(!this._checkDep('next', true) && !this._argv.quiet)) return [3 /*break*/, 2];
                        return [4 /*yield*/, inquirer_1.prompt({
                                message: 'There were errors during your project checkup, do you still want to continue?',
                                name: 'continue',
                                default: false,
                                type: 'confirm',
                            })];
                    case 1:
                        userAnswers = _a.sent();
                        _a.label = 2;
                    case 2:
                        Logging_1.nl();
                        if (!userAnswers['continue']) {
                            throw new Error('Please install the required dependencies to continue.');
                        }
                        this._shouldConfigure = Promise.resolve({ nextjs: true });
                        // eslint-disable-next-line @typescript-eslint/unbound-method
                        return [2 /*return*/, this.shouldConfigure];
                }
            });
        });
    };
    NextJs.prototype._createNextConfig = function (configDirectory, dsn) {
        var templates = fs.readdirSync(configDirectory);
        for (var _i = 0, templates_1 = templates; _i < templates_1.length; _i++) {
            var template = templates_1[_i];
            this._setTemplate(configDirectory, template, dsn);
        }
        Logging_1.red('⚠ Performance monitoring is enabled capturing 100% of transactions.\n' +
            '  Learn more in https://docs.sentry.io/product/performance/');
        Logging_1.nl();
    };
    NextJs.prototype._setTemplate = function (configDirectory, template, dsn) {
        var templatePath = path.join(configDirectory, template);
        var mergeableFile = MERGEABLE_CONFIG_PREFIX + template;
        if (!fs.existsSync(template)) {
            this._fillAndCopyTemplate(templatePath, template, dsn);
        }
        else if (!fs.existsSync(mergeableFile)) {
            this._fillAndCopyTemplate(templatePath, mergeableFile, dsn);
            Logging_1.red("File " + template + " already exists, so created " + mergeableFile + ".\n" +
                "Please, merge those files.");
            Logging_1.nl();
        }
        else {
            Logging_1.red("File " + template + " already exists, and " + mergeableFile + " also exists.\n" +
                'Please, merge those files.');
            Logging_1.nl();
        }
    };
    NextJs.prototype._fillAndCopyTemplate = function (sourcePath, targetPath, dsn) {
        var templateContent = fs.readFileSync(sourcePath).toString();
        var filledTemplate = templateContent.replace('___DSN___', dsn);
        fs.writeFileSync(targetPath, filledTemplate);
    };
    NextJs.prototype._checkDep = function (packageName, minVersion) {
        var depVersion = parseInt(_.get(appPackage, ['dependencies', packageName], '0').replace(/\D+/g, ''), 10);
        var devDepVersion = parseInt(_.get(appPackage, ['devDependencies', packageName], '0').replace(/\D+/g, ''), 10);
        var parsedVersion = parseInt(MIN_NEXTJS_VERSION.replace(/\D+/g, ''));
        if (!_.get(appPackage, "dependencies." + packageName, false) &&
            !_.get(appPackage, "devDependencies." + packageName, false)) {
            Logging_1.red("\u2717 " + packageName + " isn't in your dependencies");
            Logging_1.red("  please install it with yarn/npm");
            return false;
        }
        else if (
        // When the Next.js dependency is `latest` the int parsing will output
        // in a NaN. Don't support this.
        isNaN(depVersion) ||
            isNaN(devDepVersion)) {
            Logging_1.red("✗ `latest` version for NextJS isn't supported. Please specify a version number for `next` in your `package.json`.");
            Logging_1.nl();
            return false;
        }
        else if (minVersion &&
            depVersion < parsedVersion &&
            devDepVersion < parsedVersion) {
            Logging_1.red("\u2717 Your installed version of `" + packageName + "` is not supported, >" + MIN_NEXTJS_VERSION + " needed");
            return false;
        }
        else {
            if (minVersion) {
                Logging_1.green("\u2713 " + packageName + " > " + MIN_NEXTJS_VERSION + " is installed");
            }
            else {
                Logging_1.green("\u2713 " + packageName + " is installed");
            }
            return true;
        }
    };
    return NextJs;
}(BaseIntegration_1.BaseIntegration));
exports.NextJs = NextJs;
