// @ant-design/react-native/es/wing-blank/index.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { View } from 'react-native';
import { WithTheme } from '../style';

var WingBlank = /*#__PURE__*/function (_React$Component) {
  _inherits(WingBlank, _React$Component);

  var _super = _createSuper(WingBlank);

  function WingBlank() {
    _classCallCheck(this, WingBlank);

    return _super.apply(this, arguments);
  }

  _createClass(WingBlank, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          style = _this$props.style,
          children = _this$props.children;
      return /*#__PURE__*/React.createElement(WithTheme, null, function (_, theme) {
        var margin = theme["h_spacing_".concat(size)];
        return /*#__PURE__*/React.createElement(View, {
          style: [{
            marginLeft: margin,
            marginRight: margin
          }, style]
        }, children);
      });
    }
  }]);

  return WingBlank;
}(React.Component);

WingBlank.defaultProps = {
  size: 'lg'
};
export default WingBlank;
