// @ant-design/react-native/es/picker/cascader/Popup.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import PopupPicker from '../Popup';

var PopupCascader = /*#__PURE__*/function (_React$Component) {
  _inherits(PopupCascader, _React$Component);

  var _super = _createSuper(PopupCascader);

  function PopupCascader() {
    var _this;

    _classCallCheck(this, PopupCascader);

    _this = _super.apply(this, arguments);

    _this.onOk = function (v) {
      var _this$props = _this.props,
          onChange = _this$props.onChange,
          onOk = _this$props.onOk;

      if (onChange) {
        onChange(v);
      }

      if (onOk) {
        onOk(v);
      }
    };

    return _this;
  }

  _createClass(PopupCascader, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(PopupPicker, _extends({
        picker: this.props.cascader
      }, this.props, {
        onOk: this.onOk
      }));
    }
  }]);

  return PopupCascader;
}(React.Component);

PopupCascader.defaultProps = {
  pickerValueProp: 'value',
  pickerValueChangeProp: 'onChange'
};
export default PopupCascader;
