// @ant-design/react-native/lib/badge/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var Badge = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Badge, _React$Component);

  var _super = (0, _createSuper2["default"])(Badge);

  function Badge() {
    (0, _classCallCheck2["default"])(this, Badge);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(Badge, [{
    key: "render",
    value: function render() {
      // tslint:disable:prefer-const
      var _a = this.props,
          styles = _a.styles,
          style = _a.style,
          children = _a.children,
          text = _a.text,
          size = _a.size,
          overflowCount = _a.overflowCount,
          dot = _a.dot,
          corner = _a.corner,
          restProps = __rest(_a, ["styles", "style", "children", "text", "size", "overflowCount", "dot", "corner"]);

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        themeStyles: _index["default"],
        styles: this.props.styles
      }, function (s) {
        text = typeof text === 'number' && text > overflowCount ? "".concat(overflowCount, "+") : text; // dot mode don't need text

        if (dot) {
          text = '';
        } // fake styles


        var fakeStyles = s;
        var badgeCls = corner ? 'textCorner' : 'textDom';
        var contentDom = !dot ? /*#__PURE__*/_react["default"].createElement(_reactNative.View, (0, _extends2["default"])({}, restProps, {
          style: [s[badgeCls], fakeStyles["".concat(badgeCls).concat(size)]]
        }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [s.text]
        }, text)) : /*#__PURE__*/_react["default"].createElement(_reactNative.View, (0, _extends2["default"])({}, restProps, {
          style: [s.dot, fakeStyles["dotSize".concat(size)]]
        }));
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [s.wrap, style]
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [fakeStyles["".concat(badgeCls, "Wrap")]]
        }, children, text || dot ? contentDom : null));
      });
    }
  }]);
  return Badge;
}(_react["default"].Component);

exports["default"] = Badge;
Badge.defaultProps = {
  size: 'small',
  overflowCount: 99,
  dot: false,
  corner: false
};
