// @ant-design/react-native/lib/date-picker/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
// const now = new Date();
var PopupExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(PopupExample, _React$Component);

  var _super = (0, _createSuper2["default"])(PopupExample);

  function PopupExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, PopupExample);
    _this = _super.call(this, props);

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.state = {
      value: undefined
    };
    return _this;
  }

  (0, _createClass2["default"])(PopupExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, null, /*#__PURE__*/_react["default"].createElement(_.List, null, /*#__PURE__*/_react["default"].createElement(_.DatePicker, {
        value: this.state.value,
        mode: "date",
        defaultDate: new Date(),
        minDate: new Date(2015, 7, 6),
        maxDate: new Date(2026, 11, 3),
        onChange: this.onChange,
        format: "YYYY-MM-DD"
      }, /*#__PURE__*/_react["default"].createElement(_.List.Item, {
        arrow: "horizontal"
      }, "Select Date"))));
    }
  }]);
  return PopupExample;
}(_react["default"].Component);

exports["default"] = PopupExample;
