// @ant-design/react-native/lib/stepper/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _InputNumber = _interopRequireDefault(require("./InputNumber"));

var _style2 = _interopRequireDefault(require("./style"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var Stepper = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Stepper, _React$Component);

  var _super = (0, _createSuper2["default"])(Stepper);

  function Stepper() {
    (0, _classCallCheck2["default"])(this, Stepper);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(Stepper, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          inputStyle = _a.inputStyle,
          restProps = __rest(_a, ["inputStyle"]);

      var keyboardType = _reactNative.Platform.OS === 'android' ? 'numeric' : 'numbers-and-punctuation';
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _style2["default"]
      }, function (styles) {
        return /*#__PURE__*/_react["default"].createElement(_InputNumber["default"], (0, _extends2["default"])({}, restProps, {
          styles: styles,
          keyboardType: keyboardType,
          inputStyle: inputStyle
        }));
      });
    }
  }]);
  return Stepper;
}(_react["default"].Component);

exports["default"] = Stepper;
Stepper.defaultProps = {
  step: 1,
  readOnly: false,
  disabled: false,
  inputStyle: {}
};
