// @ant-design/react-native/lib/search-bar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

var _getLocale = require("../_util/getLocale");

var _PropsType = require("./PropsType");

var _index = _interopRequireDefault(require("./style/index"));

var _localeProvider = require("../locale-provider");

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var SearchBar = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(SearchBar, _React$Component);

  var _super = (0, _createSuper2["default"])(SearchBar);

  function SearchBar(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, SearchBar);
    _this = _super.call(this, props);

    _this.onSubmit = function (_) {
      if (_this.props.onSubmit) {
        _this.props.onSubmit(_this.state.value || '');
      }
    };

    _this.onChangeText = function (value) {
      if (!('value' in _this.props)) {
        _this.setState({
          value: value
        });
      }

      if (_this.props.onChange) {
        _this.props.onChange(value);
      }
    };

    _this.onCancel = function () {
      if (_this.props.onCancel) {
        _this.props.onCancel(_this.state.value || '');
      }
    };

    _this.onFocus = function (e) {
      _this.setState({
        focus: true
      });

      if (_this.props.onFocus) {
        _this.props.onFocus(e);
      }
    };

    _this.onBlur = function (e) {
      _this.setState({
        focus: false
      });

      if (_this.props.onBlur) {
        _this.props.onBlur(e);
      }
    };

    var value;

    if ('value' in props) {
      value = props.value;
    } else if ('defaultValue' in props) {
      value = props.defaultValue;
    } else {
      value = '';
    }

    _this.state = {
      value: value,
      focus: false
    };
    return _this;
  }

  (0, _createClass2["default"])(SearchBar, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if ('value' in nextProps) {
        this.setState({
          value: nextProps.value
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _a = this.props,
          showCancelButton = _a.showCancelButton,
          styles = _a.styles,
          propsValue = _a.value,
          cancelText = _a.cancelText,
          onChangeText = _a.onChangeText,
          onChange = _a.onChange,
          disabled = _a.disabled,
          style = _a.style,
          restProps = __rest(_a, ["showCancelButton", "styles", "value", "cancelText", "onChangeText", "onChange", "disabled", "style"]); // tslint:disable-next-line:variable-name


      var _locale = (0, _getLocale.getComponentLocale)(this.props, this.context, 'SearchBar', function () {
        return require('./locale/zh_CN');
      });

      var _this$state = this.state,
          value = _this$state.value,
          focus = _this$state.focus; // tslint:disable-next-line:variable-name

      var _showCancelButton = showCancelButton || focus;

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: styles,
        themeStyles: _index["default"]
      }, function (_styles) {
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: _styles.wrapper
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: _styles.inputWrapper
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.TextInput, (0, _extends2["default"])({
          clearButtonMode: "always",
          underlineColorAndroid: "transparent",
          editable: !disabled
        }, restProps, {
          style: [_styles.input, style],
          ref: function ref(el) {
            return _this2.inputRef = el;
          },
          value: value,
          onChangeText: _this2.onChangeText,
          onSubmitEditing: _this2.onSubmit,
          onFocus: _this2.onFocus,
          onBlur: _this2.onBlur
        }))), /*#__PURE__*/_react["default"].createElement(_icon["default"], {
          name: "search",
          style: _styles.search
        }), _showCancelButton && /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: _styles.cancelTextContainer
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: _styles.cancelText,
          onPress: _this2.onCancel
        }, cancelText || _locale.cancelText)));
      });
    }
  }]);
  return SearchBar;
}(_react["default"].Component);

exports["default"] = SearchBar;
SearchBar.defaultProps = (0, _extends2["default"])({}, _PropsType.defaultProps);
SearchBar.contextType = _localeProvider.LocaleContext;
