// @ant-design/react-native/es/flex/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { ScrollView, Text, TouchableWithoutFeedback, View } from 'react-native';
import { Button, Flex, WhiteSpace, WingBlank } from '../../';

var Circle = function Circle(props) {
  var size = props.size || 20;
  var style = {
    borderRadius: size / 2,
    backgroundColor: '#527fe4',
    width: size,
    height: size,
    margin: 1
  };
  return /*#__PURE__*/React.createElement(View, {
    style: style
  });
};

var FlexExample = /*#__PURE__*/function (_React$Component) {
  _inherits(FlexExample, _React$Component);

  var _super = _createSuper(FlexExample);

  function FlexExample() {
    _classCallCheck(this, FlexExample);

    return _super.apply(this, arguments);
  }

  _createClass(FlexExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(ScrollView, {
        style: {
          flex: 1
        },
        automaticallyAdjustContentInsets: false,
        showsHorizontalScrollIndicator: false,
        showsVerticalScrollIndicator: false
      }, /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 20,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginBottom: 10
        }
      }, "\u9879\u76EE\u7684\u6392\u5217\u65B9\u5411"), /*#__PURE__*/React.createElement(Text, null, "direction=\"row\":\u4E3B\u8F74\u4E3A\u6C34\u5E73\u65B9\u5411\uFF0C\u8D77\u70B9\u5728\u5DE6\u7AEF")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingLeft: 4,
          paddingRight: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE1")), /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingLeft: 4,
          paddingRight: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE2")), /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingLeft: 4,
          paddingRight: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE3")))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "direction=\"column\":\u4E3B\u8F74\u4E3A\u5782\u76F4\u65B9\u5411\uFF0C\u8D77\u70B9\u5728\u4E0A\u6CBF")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        direction: "column"
      }, /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingBottom: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE1")), /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingBottom: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE2")), /*#__PURE__*/React.createElement(Flex.Item, {
        style: {
          paddingBottom: 4
        }
      }, /*#__PURE__*/React.createElement(Button, {
        size: "small"
      }, "\u6309\u94AE3")))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginTop: 20,
          marginBottom: 20
        }
      }, "\u9879\u76EE\u5728\u4E3B\u8F74\u4E0A\u7684\u5BF9\u9F50\u65B9\u5F0F"), /*#__PURE__*/React.createElement(Text, null, "justify=\"start\":\u5DE6\u5BF9\u9F50")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        justify: "start"
      }, /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "justify=\"center\":\u5C45\u4E2D")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        justify: "center"
      }, /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "justify=\"end\":\u53F3\u5BF9\u9F50")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        justify: "end"
      }, /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "justify=\"between\":\u4E24\u7AEF\u5BF9\u9F50\uFF0C\u9879\u76EE\u4E4B\u95F4\u7684\u95F4\u9694\u90FD\u76F8\u7B49")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        justify: "between"
      }, /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "justify=\"around\":\u6BCF\u4E2A\u9879\u76EE\u4E24\u4FA7\u7684\u95F4\u9694\u76F8\u7B49")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        justify: "around"
      }, /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null), /*#__PURE__*/React.createElement(Circle, null))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginTop: 20,
          marginBottom: 20
        }
      }, "\u9879\u76EE\u5728\u4EA4\u53C9\u8F74\u4E0A\u7684\u5BF9\u9F50\u65B9\u5F0F"), /*#__PURE__*/React.createElement(Text, null, "align=\"start\":\u4EA4\u53C9\u8F74\u7684\u8D77\u70B9\u5BF9\u9F50")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        align: "start",
        style: {
          height: 30
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 20,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 18,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 16,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 14,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "align=\"center\":\u4EA4\u53C9\u8F74\u7684\u4E2D\u70B9\u5BF9\u9F50")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        align: "center",
        style: {
          height: 30
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 20,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 18,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 16,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 14,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "align=\"end\":\u4EA4\u53C9\u8F74\u7684\u7EC8\u70B9\u5BF9\u9F50")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        align: "end",
        style: {
          height: 30
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 20,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 18,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 16,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 14,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "align=\"stretch\":\u5982\u679C\u9879\u76EE\u672A\u8BBE\u7F6E\u9AD8\u5EA6\u6216\u8BBE\u4E3Aauto\uFF0C\u5C06\u5360\u6EE1\u6574\u4E2A\u5BB9\u5668\u7684\u9AD8\u5EA6")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(WingBlank, null, /*#__PURE__*/React.createElement(Flex, {
        align: "stretch",
        style: {
          height: 70
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 20,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 18,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 16,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C"), /*#__PURE__*/React.createElement(Text, {
        style: {
          fontSize: 14,
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: '#527fe4'
        }
      }, "\u515C\u515C")))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginBottom: 10
        }
      }, "\u662F\u5426\u6298\u884C"), /*#__PURE__*/React.createElement(Text, null, "wrap=\"wrap\":\u6362\u884C")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(TouchableWithoutFeedback, {
        onPress: function onPress() {
          return {};
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        wrap: "wrap"
      }, 'ooooooooooooooooooooooooooooo'.split('').map(function (char, i) {
        return /*#__PURE__*/React.createElement(Circle, {
          key: "".concat(i, "-").concat(char)
        });
      })))), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginTop: 5,
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Text, null, "wrap=\"nowrap\":\u4E0D\u6362\u884C")), /*#__PURE__*/React.createElement(WingBlank, {
        style: {
          marginBottom: 5
        }
      }, /*#__PURE__*/React.createElement(Flex, {
        wrap: "nowrap",
        onPress: function onPress() {
          return {};
        }
      }, 'ooooooooooooooooooooooooooooo'.split('').map(function (char, i) {
        return /*#__PURE__*/React.createElement(Circle, {
          key: "".concat(i, "-").concat(char)
        });
      }))), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(WhiteSpace, null));
    }
  }]);

  return FlexExample;
}(React.Component);

export { FlexExample as default };
