// @ant-design/react-native/lib/carousel/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var BasicCarouselExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicCarouselExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicCarouselExample);

  function BasicCarouselExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicCarouselExample);
    _this = _super.call(this, props);

    _this.onHorizontalSelectedIndexChange = function (index) {
      /* tslint:disable: no-console */
      console.log('horizontal change to', index);

      _this.setState({
        selectedIndex: index
      });
    };

    _this.state = {
      selectedIndex: 2,
      autoplay: true
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicCarouselExample, [{
    key: "onVerticalSelectedIndexChange",
    value: function onVerticalSelectedIndexChange(index) {
      /* tslint:disable: no-console */
      console.log('vertical change to', index);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingHorizontal: 15
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "horizontal"), /*#__PURE__*/_react["default"].createElement(_.Carousel, {
        style: styles.wrapper,
        selectedIndex: this.state.selectedIndex,
        autoplay: true,
        infinite: true,
        afterChange: this.onHorizontalSelectedIndexChange,
        ref: function ref(_ref) {
          return _this2.carousel = _ref;
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'red'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 1")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'blue'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 2")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'yellow'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 3")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'aqua'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 4")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'fuchsia'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 5"))), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          return _this2.carousel && _this2.carousel.goTo(0);
        }
      }, "Go to 0")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingHorizontal: 15
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "vertical"), /*#__PURE__*/_react["default"].createElement(_.Carousel, {
        style: styles.wrapper,
        selectedIndex: 1,
        autoplay: this.state.autoplay,
        infinite: true,
        afterChange: this.onVerticalSelectedIndexChange,
        vertical: true
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerVertical, {
          backgroundColor: 'red'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 1")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerVertical, {
          backgroundColor: 'blue'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 2")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerVertical, {
          backgroundColor: 'yellow'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 3")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerVertical, {
          backgroundColor: 'aqua'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 4")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [styles.containerVertical, {
          backgroundColor: 'fuchsia'
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel 5"))), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          return _this2.setState({
            autoplay: !_this2.state.autoplay
          });
        }
      }, "Toggle autoplay ", this.state.autoplay ? 'true' : 'false')));
    }
  }]);
  return BasicCarouselExample;
}(_react["default"].Component);

exports["default"] = BasicCarouselExample;

var styles = _reactNative.StyleSheet.create({
  wrapper: {
    backgroundColor: '#fff',
    width: '100%',
    height: 150
  },
  containerHorizontal: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 150
  },
  containerVertical: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 150
  },
  text: {
    color: '#fff',
    fontSize: 36
  }
});
