// @ant-design/react-native/es/pagination/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Pagination, WhiteSpace, WingBlank } from '../../';
var locale = {
  prevText: '上一步',
  nextText: '下一步'
};

var BasicPaginationExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicPaginationExample, _React$Component);

  var _super = _createSuper(BasicPaginationExample);

  function BasicPaginationExample() {
    _classCallCheck(this, BasicPaginationExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicPaginationExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(WingBlank, {
        size: "lg"
      }, /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Pagination, {
        total: 5,
        current: 1,
        locale: locale
      }), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Pagination, {
        simple: true,
        total: 5,
        current: 1,
        locale: locale
      }), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Pagination, {
        mode: "number",
        total: 5,
        current: 3
      }), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Pagination, {
        mode: "pointer",
        total: 5,
        current: 2
      }));
    }
  }]);

  return BasicPaginationExample;
}(React.Component);

export { BasicPaginationExample as default };
