// @ant-design/react-native/lib/activity-indicator/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var RNActivityIndicator = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(RNActivityIndicator, _React$Component);

  var _super = (0, _createSuper2["default"])(RNActivityIndicator);

  function RNActivityIndicator() {
    (0, _classCallCheck2["default"])(this, RNActivityIndicator);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(RNActivityIndicator, [{
    key: "_renderToast",
    value: function _renderToast() {
      var _this = this;

      var _this$props = this.props,
          _this$props$color = _this$props.color,
          color = _this$props$color === void 0 ? 'white' : _this$props$color,
          _this$props$size = _this$props.size,
          size = _this$props$size === void 0 ? 'large' : _this$props$size;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.container]
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.innerContainer, {
            height: 89
          }]
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.wrapper]
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.ActivityIndicator, {
          color: color,
          size: size
        }), _this.props.text && /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles.toast]
        }, _this.props.text))));
      });
    }
  }, {
    key: "_renderSpinner",
    value: function _renderSpinner() {
      var _this$props2 = this.props,
          color = _this$props2.color,
          size = _this$props2.size,
          text = _this$props2.text;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.spinner
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.ActivityIndicator, {
          color: color,
          size: size
        }), text && /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles.tip]
        }, text));
      });
    }
  }, {
    key: "render",
    value: function render() {
      if (this.props.animating) {
        return this.props.toast ? this._renderToast() : this._renderSpinner();
      }

      return null;
    }
  }]);
  return RNActivityIndicator;
}(_react["default"].Component);

exports["default"] = RNActivityIndicator;
RNActivityIndicator.defaultProps = {
  animating: true,
  color: 'gray',
  size: 'small',
  toast: false
};
