// @ant-design/react-native/lib/modal/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var BasicModalExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicModalExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicModalExample);

  function BasicModalExample(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, BasicModalExample);
    _this = _super.call(this, props);

    _this.onClose = function () {
      _this.setState({
        visible: false
      });
    };

    _this.onClose1 = function () {
      _this.setState({
        visible1: false
      });
    };

    _this.onClose2 = function () {
      _this.setState({
        visible2: false
      });
    };

    _this.onButtonClick = function () {
      _.Modal.alert('Title', 'alert content', [{
        text: 'Cancel',
        onPress: function onPress() {
          return console.log('cancel');
        },
        style: 'cancel'
      }, {
        text: 'OK',
        onPress: function onPress() {
          return console.log('ok');
        }
      }]);
    };

    _this.onButtonClick2 = function () {
      _.Modal.operation([{
        text: '标为未读',
        onPress: function onPress() {
          return console.log('标为未读被点击了');
        }
      }, {
        text: '置顶聊天',
        onPress: function onPress() {
          return console.log('置顶聊天被点击了');
        }
      }]);
    };

    _this.onButtonClick3 = function () {
      _.Modal.prompt('Login', 'Pleas input login information', function (login, password) {
        return console.log("login: ".concat(login, ", password: ").concat(password));
      }, 'login-password', '', ['Please input name', 'Please input password']);
    };

    _this.onButtonClick4 = function () {
      _.Modal.prompt('Input password', 'password message', function (password) {
        return console.log("password: ".concat(password));
      }, 'secure-text', 'defaultValue');
    };

    _this.onButtonClick5 = function () {
      _.Modal.prompt('Name', 'name message', function (password) {
        return console.log("password: ".concat(password));
      }, 'default', '', ['please input name']);
    };

    _this.onButtonClick6 = function () {
      _.Modal.operation([{
        text: '标为未读',
        onPress: function onPress() {
          return console.log('标为未读被点击了');
        }
      }, {
        text: '置顶聊天',
        onPress: function onPress() {
          return console.log('置顶聊天被点击了');
        }
      }], function () {
        console.log('返回键点击');
        return false;
      });
    };

    _this.state = {
      visible: false,
      visible1: false,
      visible2: false
    };
    return _this;
  }

  (0, _createClass2["default"])(BasicModalExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var footerButtons = [{
        text: 'Cancel',
        onPress: function onPress() {
          return console.log('cancel');
        }
      }, {
        text: 'Ok',
        onPress: function onPress() {
          return console.log('ok');
        }
      }];
      return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
        style: {
          marginTop: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_.WingBlank, null, /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          return _this2.setState({
            visible: true
          });
        }
      }, "showModal"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          return _this2.setState({
            visible1: true
          });
        }
      }, "transparent:false"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: function onPress() {
          return _this2.setState({
            visible2: true
          });
        }
      }, "popup"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick
      }, "Modal.alert"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick2
      }, "Modal.opertation"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick6
      }, "Modal.opertation (onBackHandler)"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick5
      }, "Modal.prompt (default)"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick3
      }, "Modal.prompt (login-password)"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Button, {
        onPress: this.onButtonClick4
      }, "Modal.prompt (secure-text)")), /*#__PURE__*/_react["default"].createElement(_.Modal, {
        title: "Title",
        transparent: true,
        onClose: this.onClose,
        maskClosable: true,
        visible: this.state.visible,
        closable: true,
        footer: footerButtons
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingVertical: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content..."), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content...")), /*#__PURE__*/_react["default"].createElement(_.Button, {
        type: "primary",
        onPress: this.onClose
      }, "close modal")), /*#__PURE__*/_react["default"].createElement(_.Modal, {
        transparent: false,
        visible: this.state.visible1,
        animationType: "slide-up",
        onClose: this.onClose1
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingVertical: 220
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content..."), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content...")), /*#__PURE__*/_react["default"].createElement(_.Button, {
        type: "primary",
        onPress: function onPress() {
          return _.Toast.info('Hello Toast in Modal now works');
        }
      }, "Hello Toast in Modal now works"), /*#__PURE__*/_react["default"].createElement(_.Button, {
        type: "primary",
        onPress: this.onClose1
      }, "close modal")), /*#__PURE__*/_react["default"].createElement(_.Modal, {
        popup: true,
        visible: this.state.visible2,
        animationType: "slide-up",
        onClose: this.onClose2
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingVertical: 20,
          paddingHorizontal: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content..."), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          textAlign: 'center'
        }
      }, "Content...")), /*#__PURE__*/_react["default"].createElement(_.Button, {
        type: "primary",
        onPress: this.onClose2
      }, "close modal")));
    }
  }]);
  return BasicModalExample;
}(_react["default"].Component);

exports["default"] = BasicModalExample;
