// @ant-design/react-native/es/list-view/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var __rest = this && this.__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

import { UltimateListView } from '@bang88/react-native-ultimate-listview';
import React from 'react';
import { ActivityIndicator, Platform, Text, View } from 'react-native';
import Icon from '../icon';
import { getComponentLocale } from '../_util/getLocale';
import zh_CN from './locale/zh_CN';
import { LocaleContext } from '../locale-provider';

var ListView = /*#__PURE__*/function (_React$PureComponent) {
  _inherits(ListView, _React$PureComponent);

  var _super = _createSuper(ListView);

  function ListView() {
    var _this;

    _classCallCheck(this, ListView);

    _this = _super.apply(this, arguments);

    _this.refresh = function () {
      if (_this.ulv) {
        _this.ulv.refresh();
      }
    };

    return _this;
  }

  _createClass(ListView, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _a = this.props,
          renderItem = _a.renderItem,
          props = __rest(_a, ["renderItem"]); // tslint:disable-next-line:variable-name


      var locale = getComponentLocale(this.props, this.context, 'ListView', function () {
        return zh_CN;
      });
      return /*#__PURE__*/React.createElement(UltimateListView, _extends({
        key: "ant-list-view",
        keyExtractor: function keyExtractor(_, index) {
          return "item-".concat(index);
        },
        refreshableMode: Platform.OS === 'ios' ? 'advanced' : 'basic',
        numColumns: 1,
        waitingSpinnerText: locale.loading,
        allLoadedText: locale.done,
        refreshableTitlePull: locale.refreshableTitlePull,
        refreshableTitleRelease: locale.refreshableTitleRelease,
        refreshableTitleRefreshing: locale.refreshableTitleRefreshing,
        emptyView: function emptyView() {
          return _this2.props.emptyView ? _this2.props.emptyView() : /*#__PURE__*/React.createElement(View, {
            style: {
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center'
            }
          }, /*#__PURE__*/React.createElement(Text, {
            style: {
              textAlign: 'center',
              padding: '10%'
            }
          }, locale.noData));
        },
        customRefreshView: function customRefreshView(status) {
          return /*#__PURE__*/React.createElement(View, {
            style: {
              flexDirection: 'row'
            }
          }, status === 2 ? /*#__PURE__*/React.createElement(ActivityIndicator, null) : /*#__PURE__*/React.createElement(Icon, {
            name: status === 0 ? 'arrow-down' : 'arrow-up',
            size: 18
          }), /*#__PURE__*/React.createElement(Text, {
            style: {
              marginLeft: 5
            }
          }, status === 0 ? locale.refreshableTitlePull : status === 1 ? locale.refreshableTitleRelease : locale.refreshableTitleRefreshing));
        }
      }, props, {
        item: renderItem,
        ref: function ref(_ref) {
          return _this2.ulv = _ref;
        }
      }));
    }
  }]);

  return ListView;
}(React.PureComponent);

ListView.contextType = LocaleContext;
export default ListView;
