// @ant-design/react-native/es/list/ListItem.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var __rest = this && this.__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

import React from 'react';
import { Image, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import Icon from '../icon';
import { WithTheme } from '../style';
import ListStyles from './style/index';
export var Brief = /*#__PURE__*/function (_React$Component) {
  _inherits(Brief, _React$Component);

  var _super = _createSuper(Brief);

  function Brief() {
    _classCallCheck(this, Brief);

    return _super.apply(this, arguments);
  }

  _createClass(Brief, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          style = _this$props.style,
          wrap = _this$props.wrap;
      var numberOfLines = {};

      if (wrap === false) {
        numberOfLines = {
          numberOfLines: 1
        };
      }

      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: ListStyles
      }, function (styles) {
        return /*#__PURE__*/React.createElement(View, {
          style: [styles.Brief]
        }, /*#__PURE__*/React.createElement(Text, _extends({
          style: [styles.BriefText, style]
        }, numberOfLines), children));
      });
    }
  }]);

  return Brief;
}(React.Component);

var Item = /*#__PURE__*/function (_React$Component2) {
  _inherits(Item, _React$Component2);

  var _super2 = _createSuper(Item);

  function Item() {
    _classCallCheck(this, Item);

    return _super2.apply(this, arguments);
  }

  _createClass(Item, [{
    key: "render",
    value: function render() {
      var _this = this;

      var _a = this.props,
          styles = _a.styles,
          children = _a.children,
          multipleLine = _a.multipleLine,
          thumb = _a.thumb,
          extra = _a.extra,
          arrow = _a.arrow,
          style = _a.style,
          onPress = _a.onPress,
          onPressIn = _a.onPressIn,
          onPressOut = _a.onPressOut,
          onLongPress = _a.onLongPress,
          delayLongPress = _a.delayLongPress,
          wrap = _a.wrap,
          disabled = _a.disabled,
          align = _a.align,
          restProps = __rest(_a, ["styles", "children", "multipleLine", "thumb", "extra", "arrow", "style", "onPress", "onPressIn", "onPressOut", "onLongPress", "delayLongPress", "wrap", "disabled", "align"]);

      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: styles,
        themeStyles: ListStyles
      }, function (itemStyles) {
        var numberOfLines = {};

        if (wrap === false) {
          numberOfLines = {
            numberOfLines: 1
          };
        }

        var underlayColor = {};

        if (!disabled && onPress) {
          underlayColor = {
            underlayColor: StyleSheet.flatten(itemStyles.underlayColor).backgroundColor,
            activeOpacity: 0.5
          };
        } else {
          underlayColor = {
            activeOpacity: 1
          };
        }

        var alignStyle = {};

        if (align === 'top') {
          alignStyle = {
            alignItems: 'flex-start'
          };
        } else if (align === 'bottom') {
          alignStyle = {
            alignItems: 'flex-end'
          };
        }

        var contentDom;

        if (Array.isArray(children)) {
          var tempContentDom = [];
          children.forEach(function (el, index) {
            if ( /*#__PURE__*/React.isValidElement(el)) {
              tempContentDom.push(el);
            } else {
              tempContentDom.push( /*#__PURE__*/React.createElement(Text, _extends({
                style: [itemStyles.Content]
              }, numberOfLines, {
                key: "".concat(index, "-children")
              }), el));
            }
          });
          contentDom = /*#__PURE__*/React.createElement(View, {
            style: [itemStyles.column]
          }, tempContentDom);
        } else {
          if (children && /*#__PURE__*/React.isValidElement(children)) {
            contentDom = /*#__PURE__*/React.createElement(View, {
              style: [itemStyles.column]
            }, children);
          } else {
            contentDom = /*#__PURE__*/React.createElement(View, {
              style: [itemStyles.column]
            }, /*#__PURE__*/React.createElement(Text, _extends({
              style: [itemStyles.Content]
            }, numberOfLines), children));
          }
        }

        var extraDom;

        if (extra) {
          extraDom = /*#__PURE__*/React.createElement(View, {
            style: [itemStyles.column]
          }, /*#__PURE__*/React.createElement(Text, _extends({
            style: [itemStyles.Extra]
          }, numberOfLines), extra));

          if ( /*#__PURE__*/React.isValidElement(extra)) {
            var extraChildren = extra.props.children;

            if (Array.isArray(extraChildren)) {
              var tempExtraDom = [];
              extraChildren.forEach(function (el, index) {
                if (typeof el === 'string') {
                  tempExtraDom.push( /*#__PURE__*/React.createElement(Text, _extends({}, numberOfLines, {
                    style: [itemStyles.Extra],
                    key: "".concat(index, "-children")
                  }), el));
                } else {
                  tempExtraDom.push(el);
                }
              });
              extraDom = /*#__PURE__*/React.createElement(View, {
                style: [itemStyles.column]
              }, tempExtraDom);
            } else {
              extraDom = extra;
            }
          }
        }

        var arrEnum = {
          horizontal: /*#__PURE__*/React.createElement(Icon, {
            name: "right",
            style: itemStyles.Arrow
          }),
          down: /*#__PURE__*/React.createElement(Icon, {
            name: "down",
            style: itemStyles.ArrowV
          }),
          up: /*#__PURE__*/React.createElement(Icon, {
            name: "up",
            style: itemStyles.ArrowV
          })
        };
        var itemView = /*#__PURE__*/React.createElement(View, _extends({}, restProps, {
          style: [itemStyles.Item, style]
        }), typeof thumb === 'string' ? /*#__PURE__*/React.createElement(Image, {
          source: {
            uri: thumb
          },
          style: [itemStyles.Thumb, multipleLine && itemStyles.multipleThumb]
        }) : thumb, /*#__PURE__*/React.createElement(View, {
          style: [itemStyles.Line, multipleLine && itemStyles.multipleLine, multipleLine && alignStyle]
        }, contentDom, extraDom, arrow ? arrEnum[arrow] || /*#__PURE__*/React.createElement(View, {
          style: itemStyles.Arrow
        }) : null));
        return /*#__PURE__*/React.createElement(TouchableHighlight, _extends({}, underlayColor, {
          // TODO: fix onClick
          onPress: _this.props.onClick ? _this.props.onClick : onPress || undefined,
          onPressIn: onPressIn,
          onPressOut: onPressOut,
          onLongPress: onLongPress,
          delayLongPress: delayLongPress
        }), itemView);
      });
    }
  }]);

  return Item;
}(React.Component);

export { Item as default };
Item.defaultProps = {
  multipleLine: false,
  wrap: false,
  delayLongPress: 500,
  onLongPress: function onLongPress() {}
};
Item.Brief = Brief;
