// @ant-design/react-native/es/white-space/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import { Text, View } from 'react-native';
import { WhiteSpace } from '../../';

var PlaceHolder = function PlaceHolder(props) {
  return /*#__PURE__*/React.createElement(View, _extends({
    style: {
      backgroundColor: '#fff',
      height: 30,
      alignItems: 'center',
      justifyContent: 'center'
    }
  }, props), /*#__PURE__*/React.createElement(Text, {
    style: {
      color: '#bbb'
    }
  }, "Block"));
};

var WhiteSpaceExample = /*#__PURE__*/function (_React$Component) {
  _inherits(WhiteSpaceExample, _React$Component);

  var _super = _createSuper(WhiteSpaceExample);

  function WhiteSpaceExample() {
    _classCallCheck(this, WhiteSpaceExample);

    return _super.apply(this, arguments);
  }

  _createClass(WhiteSpaceExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "xs"
      }), /*#__PURE__*/React.createElement(PlaceHolder, null), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "sm"
      }), /*#__PURE__*/React.createElement(PlaceHolder, null), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(PlaceHolder, null), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(PlaceHolder, null), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "xl"
      }), /*#__PURE__*/React.createElement(PlaceHolder, null));
    }
  }]);

  return WhiteSpaceExample;
}(React.Component);

export { WhiteSpaceExample as default };
