// @ant-design/react-native/es/modal/PromptContainer.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, TextInput, View } from 'react-native';
import { WithTheme } from '../style';
import { getComponentLocale } from '../_util/getLocale';
import zh_CN from './locale/zh_CN';
import Modal from './Modal';
import promptStyles from './style/prompt';
import { LocaleContext } from '../locale-provider';

var PropmptContainer = /*#__PURE__*/function (_React$Component) {
  _inherits(PropmptContainer, _React$Component);

  var _super = _createSuper(PropmptContainer);

  function PropmptContainer(props) {
    var _this;

    _classCallCheck(this, PropmptContainer);

    _this = _super.call(this, props);

    _this.onBackAndroid = function () {
      var onBackHandler = _this.props.onBackHandler;

      if (typeof onBackHandler === 'function') {
        var flag = onBackHandler();

        if (flag) {
          _this.onClose();
        }

        return flag;
      }

      if (_this.state.visible) {
        _this.onClose();

        return true;
      }

      return false;
    };

    _this.onClose = function () {
      _this.setState({
        visible: false
      });
    };

    _this.state = {
      visible: true,
      text: props.defaultValue,
      password: props.type === 'secure-text' ? props.defaultValue : ''
    };
    return _this;
  }

  _createClass(PropmptContainer, [{
    key: "onChangeText",
    value: function onChangeText(type, value) {
      this.setState(_defineProperty({}, type, value));
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          title = _this$props.title,
          onAnimationEnd = _this$props.onAnimationEnd,
          message = _this$props.message,
          type = _this$props.type,
          actions = _this$props.actions,
          placeholders = _this$props.placeholders;
      var _this$state = this.state,
          text = _this$state.text,
          password = _this$state.password;

      var getArgs = function getArgs(func) {
        if (type === 'login-password') {
          return func.apply(this, [text, password]);
        } else if (type === 'secure-text') {
          return func.apply(this, [password]);
        }

        return func.apply(this, [text]);
      }; // tslint:disable-next-line:variable-name


      var _locale = getComponentLocale(this.props, this.context, 'Modal', function () {
        return zh_CN;
      });

      var callbacks;

      if (typeof actions === 'function') {
        callbacks = [{
          text: _locale.cancelText,
          style: 'cancel',
          onPress: function onPress() {}
        }, {
          text: _locale.okText,
          onPress: function onPress() {
            return getArgs(actions);
          }
        }];
      } else {
        callbacks = actions.map(function (item) {
          return {
            text: item.text,
            onPress: function onPress() {
              if (item.onPress) {
                return getArgs(item.onPress);
              }
            },
            style: item.style || {}
          };
        });
      }

      var footer = callbacks.map(function (button) {
        // tslint:disable-next-line:only-arrow-functions
        var orginPress = button.onPress || function () {};

        button.onPress = function () {
          var res = orginPress();

          if (res && res.then) {
            res.then(function () {
              _this2.onClose();
            });
          } else {
            _this2.onClose();
          }
        };

        return button;
      });
      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: promptStyles
      }, function (styles) {
        var firstStyle = [styles.inputWrapper];
        var lastStyle = [styles.inputWrapper];

        if (type === 'login-password') {
          firstStyle.push(styles.inputFirst);
          lastStyle.push(styles.inputLast);
        } else if (type === 'secure-text') {
          lastStyle.push(styles.inputFirst);
          lastStyle.push(styles.inputLast);
        } else {
          firstStyle.push(styles.inputFirst);
          firstStyle.push(styles.inputLast);
        }

        return /*#__PURE__*/React.createElement(Modal, {
          transparent: true,
          title: title,
          visible: _this2.state.visible,
          footer: footer,
          onAnimationEnd: onAnimationEnd,
          onRequestClose: _this2.onBackAndroid
        }, message ? /*#__PURE__*/React.createElement(Text, {
          style: styles.message
        }, message) : null, /*#__PURE__*/React.createElement(View, {
          style: styles.inputGroup
        }, type !== 'secure-text' && /*#__PURE__*/React.createElement(View, {
          style: firstStyle
        }, /*#__PURE__*/React.createElement(TextInput, {
          autoFocus: true,
          onChangeText: function onChangeText(value) {
            _this2.onChangeText('text', value);
          },
          value: _this2.state.text,
          style: styles.input,
          underlineColorAndroid: "transparent",
          placeholder: placeholders[0]
        })), (type === 'secure-text' || type === 'login-password') && /*#__PURE__*/React.createElement(View, {
          style: lastStyle
        }, /*#__PURE__*/React.createElement(TextInput, {
          autoFocus: true,
          secureTextEntry: true,
          onChangeText: function onChangeText(value) {
            _this2.onChangeText('password', value);
          },
          value: _this2.state.password,
          style: styles.input,
          underlineColorAndroid: "transparent",
          placeholder: placeholders[1]
        }))));
      });
    }
  }]);

  return PropmptContainer;
}(React.Component);

export { PropmptContainer as default };
PropmptContainer.defaultProps = {
  type: 'default',
  defaultValue: ''
};
PropmptContainer.contextType = LocaleContext;
