// @ant-design/react-native/es/radio/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { List, Radio, WhiteSpace } from '../../';
var RadioItem = Radio.RadioItem;

var BasicRadioExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicRadioExample, _React$Component);

  var _super = _createSuper(BasicRadioExample);

  function BasicRadioExample() {
    var _this;

    _classCallCheck(this, BasicRadioExample);

    _this = _super.apply(this, arguments);
    _this.state = {
      part1Value: 1,
      part2Value: 1
    };
    return _this;
  }

  _createClass(BasicRadioExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/React.createElement(Radio, {
        checked: this.state.part1Value === 1,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part1Value: 1
            });
          }
        },
        style: {
          borderWidth: 1,
          borderColor: '#999',
          margin: 10
        }
      }, "Support"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Radio, {
        checked: this.state.part1Value === 2,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part1Value: 2
            });
          }
        },
        style: {
          borderWidth: 1,
          borderColor: '#999',
          margin: 10
        }
      }), /*#__PURE__*/React.createElement(WhiteSpace, null)), /*#__PURE__*/React.createElement(List, {
        style: {
          marginTop: 12
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginTop: 12
        }
      }, "Form radio, radio in general list."), /*#__PURE__*/React.createElement(RadioItem, {
        checked: this.state.part2Value === 1,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part2Value: 1
            });
          }
        }
      }, "Use Ant Design Component"), /*#__PURE__*/React.createElement(RadioItem, {
        checked: this.state.part2Value === 2,
        onChange: function onChange(event) {
          if (event.target.checked) {
            _this2.setState({
              part2Value: 2
            });
          }
        }
      }, "Use Ant Design Component"), /*#__PURE__*/React.createElement(RadioItem, {
        disabled: true
      }, "Set disabled"), /*#__PURE__*/React.createElement(RadioItem, {
        disabled: true,
        checked: true
      }, "Set disabled")));
    }
  }]);

  return BasicRadioExample;
}(React.Component);

export { BasicRadioExample as default };
