// @ant-design/react-native/es/swipe-action/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { View } from 'react-native';
import { List, SwipeAction } from '../../';

var BasicSwipeActionExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicSwipeActionExample, _React$Component);

  var _super = _createSuper(BasicSwipeActionExample);

  function BasicSwipeActionExample() {
    _classCallCheck(this, BasicSwipeActionExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicSwipeActionExample, [{
    key: "render",
    value: function render() {
      var right = [{
        text: 'More',
        onPress: function onPress() {
          return console.log('more');
        },
        backgroundColor: 'orange',
        color: 'white'
      }, {
        text: 'Delete',
        onPress: function onPress() {
          return console.log('delete');
        },
        backgroundColor: 'red',
        color: 'white'
      }];
      var left = [{
        text: 'Read',
        onPress: function onPress() {
          return console.log('read');
        },
        backgroundColor: 'blue',
        color: 'white'
      }, {
        text: 'Reply',
        onPress: function onPress() {
          return console.log('reply');
        },
        backgroundColor: 'green',
        color: 'white'
      }];
      return /*#__PURE__*/React.createElement(View, {
        style: {
          paddingTop: 30
        }
      }, /*#__PURE__*/React.createElement(List, null, /*#__PURE__*/React.createElement(SwipeAction, {
        right: right,
        left: left,
        onSwipeableOpen: function onSwipeableOpen() {
          return console.log('open');
        },
        onSwipeableClose: function onSwipeableClose() {
          return console.log('close');
        }
      }, /*#__PURE__*/React.createElement(List.Item, {
        extra: "extra content"
      }, "Simple example: left and right buttons"))));
    }
  }]);

  return BasicSwipeActionExample;
}(React.Component);

export { BasicSwipeActionExample as default };
