// @ant-design/react-native/es/checkbox/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Checkbox, List, WhiteSpace } from '../../';
var AgreeItem = Checkbox.AgreeItem;
var CheckboxItem = Checkbox.CheckboxItem;

var BasicCheckboxExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicCheckboxExample, _React$Component);

  var _super = _createSuper(BasicCheckboxExample);

  function BasicCheckboxExample(props, context) {
    var _this;

    _classCallCheck(this, BasicCheckboxExample);

    _this = _super.call(this, props, context);
    _this.state = {
      checkBox1: true,
      agreeItem1: true,
      checkboxItem1: true
    };
    return _this;
  }

  _createClass(BasicCheckboxExample, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/React.createElement(Checkbox, {
        checked: this.state.checkBox1,
        style: {
          color: '#f00'
        },
        onChange: function onChange(event) {
          _this2.setState({
            checkBox1: event.target.checked
          });
        }
      }), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Checkbox, null, "Checkbox"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Checkbox, {
        checked: true,
        disabled: true
      }), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Checkbox, {
        disabled: true
      })), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(AgreeItem, null, "Agree agreement agreement agreement agreement agreement agreement agreement"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(AgreeItem, {
        checked: this.state.agreeItem1,
        checkboxStyle: {
          color: '#f00'
        },
        onChange: function onChange(event) {
          _this2.setState({
            agreeItem1: event.target.checked
          });
        }
      }, "Agree agreement"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(AgreeItem, {
        disabled: true
      }, "Not selected. Not editable"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(AgreeItem, {
        checked: true,
        disabled: true
      }, "Force selected. Not editable"), /*#__PURE__*/React.createElement(List, {
        style: {
          marginTop: 12
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginTop: 12
        }
      }, "Multiple options"), /*#__PURE__*/React.createElement(CheckboxItem, {
        checked: this.state.checkboxItem1,
        onChange: function onChange(event) {
          _this2.setState({
            checkboxItem1: event.target.checked
          });
        }
      }, "Option 1"), /*#__PURE__*/React.createElement(CheckboxItem, null, "Option 2"), /*#__PURE__*/React.createElement(CheckboxItem, {
        disabled: true
      }, "Option 3"), /*#__PURE__*/React.createElement(CheckboxItem, {
        disabled: true,
        checked: true
      }, "Option 4")));
    }
  }]);

  return BasicCheckboxExample;
}(React.Component);

export { BasicCheckboxExample as default };
