// @ant-design/react-native/es/date-picker/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { View } from 'react-native';
import { DatePicker, List } from '../../'; // const now = new Date();

var PopupExample = /*#__PURE__*/function (_React$Component) {
  _inherits(PopupExample, _React$Component);

  var _super = _createSuper(PopupExample);

  function PopupExample(props) {
    var _this;

    _classCallCheck(this, PopupExample);

    _this = _super.call(this, props);

    _this.onChange = function (value) {
      _this.setState({
        value: value
      });
    };

    _this.state = {
      value: undefined
    };
    return _this;
  }

  _createClass(PopupExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(List, null, /*#__PURE__*/React.createElement(DatePicker, {
        value: this.state.value,
        mode: "date",
        defaultDate: new Date(),
        minDate: new Date(2015, 7, 6),
        maxDate: new Date(2026, 11, 3),
        onChange: this.onChange,
        format: "YYYY-MM-DD"
      }, /*#__PURE__*/React.createElement(List.Item, {
        arrow: "horizontal"
      }, "Select Date"))));
    }
  }]);

  return PopupExample;
}(React.Component);

export { PopupExample as default };
