// @ant-design/react-native/lib/result/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var ResultExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(ResultExample, _React$Component);

  var _super = (0, _createSuper2["default"])(ResultExample);

  function ResultExample() {
    (0, _classCallCheck2["default"])(this, ResultExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(ResultExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
        style: {
          backgroundColor: '#F5F5F9',
          flex: 1
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "URI \u56FE\u7247"), /*#__PURE__*/_react["default"].createElement(_.Result, {
        imgUrl: {
          uri: 'https://zos.alipayobjects.com/rmsportal/GcBguhrOdlYvGfnsXgrE.png'
        },
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "Image source"), /*#__PURE__*/_react["default"].createElement(_.Result, {
        imgUrl: require('./alipay.png'),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1"
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "Image element"), /*#__PURE__*/_react["default"].createElement(_.Result, {
        img: /*#__PURE__*/_react["default"].createElement(_reactNative.Image, {
          source: require('./alipay.png'),
          style: {
            width: 60,
            height: 60
          }
        }),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1")
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          margin: 10,
          color: '#999'
        }
      }, "\u542B button \u64CD\u4F5C"), /*#__PURE__*/_react["default"].createElement(_.Result, {
        img: /*#__PURE__*/_react["default"].createElement(_reactNative.Image, {
          source: require('./alipay.png'),
          style: {
            width: 60,
            height: 60
          }
        }),
        title: "\u9A8C\u8BC1\u6210\u529F",
        message: "\u6240\u63D0\u4EA4\u5185\u5BB9\u5DF2\u6210\u529F\u5B8C\u6210\u9A8C\u8BC1",
        buttonText: "\u5B8C\u6210",
        buttonType: "primary",
        onButtonClick: function onButtonClick(e) {
          return alert(e.toString());
        }
      }));
    }
  }]);
  return ResultExample;
}(_react["default"].Component);

exports["default"] = ResultExample;
