// @ant-design/react-native/lib/grid/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var data = Array.from(new Array(9)).map(function (_val, i) {
  return {
    icon: 'https://os.alipayobjects.com/rmsportal/IptWdCkrtkAUfjE.png',
    text: "Name".concat(i)
  };
});

var BasicGridExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicGridExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicGridExample);

  function BasicGridExample() {
    (0, _classCallCheck2["default"])(this, BasicGridExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicGridExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Simple")), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          padding: 10
        }]
      }, /*#__PURE__*/_react["default"].createElement(_.Grid, {
        data: data,
        hasLine: false
      })), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Carousel")), /*#__PURE__*/_react["default"].createElement(_.Grid, {
        data: data,
        columnNum: 3,
        isCarousel: true,
        carouselProps: {
          style: {
            width: '100%',
            height: 320
          }
        },
        onPress: function onPress(_el, index) {
          return alert(index);
        }
      }), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: [{
          margin: 10
        }]
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "Custom GridCell Style")), /*#__PURE__*/_react["default"].createElement(_.Grid, {
        data: data,
        columnNum: 3,
        itemStyle: {
          height: 150,
          backgroundColor: '#ffff00'
        }
      }));
    }
  }]);
  return BasicGridExample;
}(_react["default"].Component);

exports["default"] = BasicGridExample;
