// @ant-design/react-native/es/carousel/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button, Carousel } from '../../';

var BasicCarouselExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicCarouselExample, _React$Component);

  var _super = _createSuper(BasicCarouselExample);

  function BasicCarouselExample(props) {
    var _this;

    _classCallCheck(this, BasicCarouselExample);

    _this = _super.call(this, props);

    _this.onHorizontalSelectedIndexChange = function (index) {
      /* tslint:disable: no-console */
      console.log('horizontal change to', index);

      _this.setState({
        selectedIndex: index
      });
    };

    _this.state = {
      selectedIndex: 2,
      autoplay: true
    };
    return _this;
  }

  _createClass(BasicCarouselExample, [{
    key: "onVerticalSelectedIndexChange",
    value: function onVerticalSelectedIndexChange(index) {
      /* tslint:disable: no-console */
      console.log('vertical change to', index);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement(View, {
        style: {
          marginTop: 30
        }
      }, /*#__PURE__*/React.createElement(View, {
        style: {
          paddingHorizontal: 15
        }
      }, /*#__PURE__*/React.createElement(Text, null, "horizontal"), /*#__PURE__*/React.createElement(Carousel, {
        style: styles.wrapper,
        selectedIndex: this.state.selectedIndex,
        autoplay: true,
        infinite: true,
        afterChange: this.onHorizontalSelectedIndexChange,
        ref: function ref(_ref) {
          return _this2.carousel = _ref;
        }
      }, /*#__PURE__*/React.createElement(View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'red'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 1")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'blue'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 2")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'yellow'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 3")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'aqua'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 4")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerHorizontal, {
          backgroundColor: 'fuchsia'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 5"))), /*#__PURE__*/React.createElement(Button, {
        onPress: function onPress() {
          return _this2.carousel && _this2.carousel.goTo(0);
        }
      }, "Go to 0")), /*#__PURE__*/React.createElement(View, {
        style: {
          paddingHorizontal: 15
        }
      }, /*#__PURE__*/React.createElement(Text, null, "vertical"), /*#__PURE__*/React.createElement(Carousel, {
        style: styles.wrapper,
        selectedIndex: 1,
        autoplay: this.state.autoplay,
        infinite: true,
        afterChange: this.onVerticalSelectedIndexChange,
        vertical: true
      }, /*#__PURE__*/React.createElement(View, {
        style: [styles.containerVertical, {
          backgroundColor: 'red'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 1")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerVertical, {
          backgroundColor: 'blue'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 2")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerVertical, {
          backgroundColor: 'yellow'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 3")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerVertical, {
          backgroundColor: 'aqua'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 4")), /*#__PURE__*/React.createElement(View, {
        style: [styles.containerVertical, {
          backgroundColor: 'fuchsia'
        }]
      }, /*#__PURE__*/React.createElement(Text, null, "Carousel 5"))), /*#__PURE__*/React.createElement(Button, {
        onPress: function onPress() {
          return _this2.setState({
            autoplay: !_this2.state.autoplay
          });
        }
      }, "Toggle autoplay ", this.state.autoplay ? 'true' : 'false')));
    }
  }]);

  return BasicCarouselExample;
}(React.Component);

export { BasicCarouselExample as default };
var styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#fff',
    width: '100%',
    height: 150
  },
  containerHorizontal: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 150
  },
  containerVertical: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 150
  },
  text: {
    color: '#fff',
    fontSize: 36
  }
});
