// @ant-design/react-native/es/button/demo/basic.js
/* tslint:disable:no-unused-variable */
import React from 'react';
import { Button, WhiteSpace, WingBlank } from '../../';
/* tslint:disable:no-console */

export default (function () {
  return /*#__PURE__*/React.createElement(WingBlank, null, /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, null, "default"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    disabled: true
  }, "default disabled"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    type: "primary"
  }, "primary"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    type: "primary",
    disabled: true
  }, "primary disabled"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    type: "warning"
  }, "warning"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    type: "warning",
    disabled: true
  }, "warning disabled"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    loading: true
  }, "loading button"), /*#__PURE__*/React.createElement(Button, {
    activeStyle: false
  }, "\u65E0\u70B9\u51FB\u53CD\u9988"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(Button, {
    activeStyle: {
      backgroundColor: 'red'
    }
  }, "custom feedback style"), /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(WingBlank, {
    style: {
      marginTop: 20,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center'
    }
  }, /*#__PURE__*/React.createElement(Button, {
    type: "ghost"
  }, "ghost"), /*#__PURE__*/React.createElement(Button, {
    type: "ghost",
    disabled: true
  }, "ghost disabled"), /*#__PURE__*/React.createElement(Button, {
    type: "ghost",
    size: "small"
  }, "ghost")));
});
