// @ant-design/react-native/lib/picker/cascader/Popup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _Popup = _interopRequireDefault(require("../Popup"));

var PopupCascader = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(PopupCascader, _React$Component);

  var _super = (0, _createSuper2["default"])(PopupCascader);

  function PopupCascader() {
    var _this;

    (0, _classCallCheck2["default"])(this, PopupCascader);
    _this = _super.apply(this, arguments);

    _this.onOk = function (v) {
      var _this$props = _this.props,
          onChange = _this$props.onChange,
          onOk = _this$props.onOk;

      if (onChange) {
        onChange(v);
      }

      if (onOk) {
        onOk(v);
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(PopupCascader, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_Popup["default"], (0, _extends2["default"])({
        picker: this.props.cascader
      }, this.props, {
        onOk: this.onOk
      }));
    }
  }]);
  return PopupCascader;
}(_react["default"].Component);

PopupCascader.defaultProps = {
  pickerValueProp: 'value',
  pickerValueChangeProp: 'onChange'
};
var _default = PopupCascader;
exports["default"] = _default;
