// @ant-design/react-native/es/card/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import React from 'react';
import { Text, View } from 'react-native';
import { Card, WhiteSpace, WingBlank } from '../../';

var BasicCardExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicCardExample, _React$Component);

  var _super = _createSuper(BasicCardExample);

  function BasicCardExample() {
    _classCallCheck(this, BasicCardExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicCardExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, {
        style: {
          paddingTop: 30
        }
      }, /*#__PURE__*/React.createElement(WingBlank, {
        size: "lg"
      }, /*#__PURE__*/React.createElement(Card, null, /*#__PURE__*/React.createElement(Card.Header, {
        title: "This is title",
        thumbStyle: {
          width: 30,
          height: 30
        },
        thumb: "https://gw.alipayobjects.com/zos/rmsportal/MRhHctKOineMbKAZslML.jpg",
        extra: "this is extra"
      }), /*#__PURE__*/React.createElement(Card.Body, null, /*#__PURE__*/React.createElement(View, {
        style: {
          height: 42
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginLeft: 16
        }
      }, "Card Content"))), /*#__PURE__*/React.createElement(Card.Footer, {
        content: "footer content",
        extra: "footer extra content"
      }))), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(Card, {
        full: true
      }, /*#__PURE__*/React.createElement(Card.Header, {
        title: "Full Column",
        thumbStyle: {
          width: 30,
          height: 30
        },
        thumb: "https://gw.alipayobjects.com/zos/rmsportal/MRhHctKOineMbKAZslML.jpg",
        extra: "this is extra"
      }), /*#__PURE__*/React.createElement(Card.Body, null, /*#__PURE__*/React.createElement(View, {
        style: {
          height: 42
        }
      }, /*#__PURE__*/React.createElement(Text, {
        style: {
          marginLeft: 16
        }
      }, "Card Content"))), /*#__PURE__*/React.createElement(Card.Footer, {
        content: "footer content",
        extra: "footer extra content"
      })));
    }
  }]);

  return BasicCardExample;
}(React.Component);

export { BasicCardExample as default };
