// @ant-design/react-native/lib/pagination/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _index = _interopRequireDefault(require("../button/index"));

var _index2 = _interopRequireDefault(require("../flex/index"));

var _style = require("../style");

var _getLocale = require("../_util/getLocale");

var _zh_CN = _interopRequireDefault(require("./locale/zh_CN"));

var _index3 = _interopRequireDefault(require("./style/index"));

var _localeProvider = require("../locale-provider");

var Pagination = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Pagination, _React$Component);

  var _super = (0, _createSuper2["default"])(Pagination);

  function Pagination(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Pagination);
    _this = _super.call(this, props);
    _this.state = {
      current: props.current
    };
    return _this;
  }

  (0, _createClass2["default"])(Pagination, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (nextProps.current !== this.state.current) {
        this.setState({
          current: nextProps.current
        });
      }
    }
  }, {
    key: "onChange",
    value: function onChange(p) {
      this.setState({
        current: p
      });

      if (this.props.onChange) {
        this.props.onChange(p);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          mode = _this$props.mode,
          total = _this$props.total,
          simple = _this$props.simple;
      var locale = (0, _getLocale.getComponentLocale)(this.props, this.context, 'Pagination', function () {
        return _zh_CN["default"];
      });
      var prevText = locale.prevText,
          nextText = locale.nextText;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index3["default"]
      }, function (styles) {
        var current = _this2.state.current;
        var simpleItem = !simple ? /*#__PURE__*/_react["default"].createElement(_index2["default"].Item, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.numberStyle]
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles.activeTextStyle]
        }, current), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: [styles.totalStyle]
        }, "/", total))) : /*#__PURE__*/_react["default"].createElement(_index2["default"].Item, null);

        var markup = /*#__PURE__*/_react["default"].createElement(_index2["default"], null, /*#__PURE__*/_react["default"].createElement(_index2["default"].Item, null, /*#__PURE__*/_react["default"].createElement(_index["default"], {
          disabled: current <= 1,
          onPress: function onPress() {
            return _this2.onChange(current - 1);
          }
        }, prevText)), simpleItem, /*#__PURE__*/_react["default"].createElement(_index2["default"].Item, null, /*#__PURE__*/_react["default"].createElement(_index["default"], {
          disabled: current >= total,
          onPress: function onPress() {
            return _this2.onChange(current + 1);
          }
        }, nextText)));

        if (mode === 'number') {
          markup = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [styles.numberStyle]
          }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.activeTextStyle]
          }, current), /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: [styles.totalStyle]
          }, "/", total));
        } else if (mode === 'pointer') {
          var arr = [];

          for (var i = 0; i < total; i++) {
            arr.push( /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
              key: "dot-".concat(i),
              // tslint:disable-next-line:jsx-no-multiline-js
              style: [styles.pointStyle, styles.spaceStyle, i + 1 === current && styles.pointActiveStyle]
            }));
          }

          markup = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: [styles.indicatorStyle, _this2.props.indicatorStyle]
          }, arr);
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.container, style]
        }, markup);
      });
    }
  }]);
  return Pagination;
}(_react["default"].Component);

exports["default"] = Pagination;
Pagination.defaultProps = {
  mode: 'button',
  current: 1,
  total: 0,
  simple: false,
  onChange: function onChange() {},
  indicatorStyle: null
};
Pagination.contextType = _localeProvider.LocaleContext;
