// @ant-design/react-native/lib/accordion/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _Accordion = _interopRequireDefault(require("react-native-collapsible/Accordion"));

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

var _index = _interopRequireDefault(require("./style/index"));

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var AccordionPanel = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(AccordionPanel, _React$Component);

  var _super = (0, _createSuper2["default"])(AccordionPanel);

  function AccordionPanel() {
    (0, _classCallCheck2["default"])(this, AccordionPanel);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(AccordionPanel, [{
    key: "render",
    value: function render() {
      return null;
    }
  }]);
  return AccordionPanel;
}(_react["default"].Component);

var Accordion = /*#__PURE__*/function (_React$Component2) {
  (0, _inherits2["default"])(Accordion, _React$Component2);

  var _super2 = (0, _createSuper2["default"])(Accordion);

  function Accordion() {
    var _this;

    (0, _classCallCheck2["default"])(this, Accordion);
    _this = _super2.apply(this, arguments);

    _this.renderHeader = function (styles) {
      return function (section, _, isActive) {
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.header, section.style]
        }, /*#__PURE__*/_react["default"].isValidElement(section.title) ? section.title : /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.headerWrap
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: styles.headerText
        }, section.title)), /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.arrow
        }, /*#__PURE__*/_react["default"].createElement(_icon["default"], {
          name: isActive ? 'up' : 'down',
          style: styles.arrow
        })));
      };
    };

    _this.renderContent = function (styles) {
      return function (section) {
        return /*#__PURE__*/_react["default"].isValidElement(section.content) ? section.content : /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: styles.content
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: styles.contentText
        }, section.content));
      };
    };

    return _this;
  }

  (0, _createClass2["default"])(Accordion, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _a = this.props,
          children = _a.children,
          style = _a.style,
          _a$activeSections = _a.activeSections,
          activeSections = _a$activeSections === void 0 ? [] : _a$activeSections,
          rest = __rest(_a, ["children", "style", "activeSections"]);

      var styles = this.props.styles;

      var headers = _react["default"].Children.map(children, function (child) {
        return {
          title: child.props.header,
          content: child.props.children,
          style: child.props.style || {}
        };
      });

      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        themeStyles: _index["default"],
        styles: styles
      }, function (s) {
        return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [s.container, style]
        }, /*#__PURE__*/_react["default"].createElement(_Accordion["default"], (0, _extends2["default"])({
          underlayColor: "transparent",
          duration: 0
        }, rest, {
          renderHeader: _this2.renderHeader(s),
          renderContent: _this2.renderContent(s),
          sections: headers,
          activeSections: activeSections
        })));
      });
    }
  }]);
  return Accordion;
}(_react["default"].Component);

Accordion.Panel = AccordionPanel;
var _default = Accordion;
exports["default"] = _default;
