// @ant-design/react-native/es/stepper/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var __rest = this && this.__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

import React from 'react';
import { Platform } from 'react-native';
import { WithTheme } from '../style';
import InputNumber from './InputNumber';
import StepperStyles from './style';

var Stepper = /*#__PURE__*/function (_React$Component) {
  _inherits(Stepper, _React$Component);

  var _super = _createSuper(Stepper);

  function Stepper() {
    _classCallCheck(this, Stepper);

    return _super.apply(this, arguments);
  }

  _createClass(Stepper, [{
    key: "render",
    value: function render() {
      var _a = this.props,
          inputStyle = _a.inputStyle,
          restProps = __rest(_a, ["inputStyle"]);

      var keyboardType = Platform.OS === 'android' ? 'numeric' : 'numbers-and-punctuation';
      return /*#__PURE__*/React.createElement(WithTheme, {
        styles: this.props.styles,
        themeStyles: StepperStyles
      }, function (styles) {
        return /*#__PURE__*/React.createElement(InputNumber, _extends({}, restProps, {
          styles: styles,
          keyboardType: keyboardType,
          inputStyle: inputStyle
        }));
      });
    }
  }]);

  return Stepper;
}(React.Component);

export { Stepper as default };
Stepper.defaultProps = {
  step: 1,
  readOnly: false,
  disabled: false,
  inputStyle: {}
};
