// @ant-design/react-native/es/wing-blank/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import { Text, View } from 'react-native';
import { WhiteSpace, WingBlank } from '../../';

var PlaceHolder = function PlaceHolder(props) {
  return /*#__PURE__*/React.createElement(View, _extends({
    style: {
      backgroundColor: '#fff',
      height: 30,
      alignItems: 'center',
      justifyContent: 'center'
    }
  }, props), /*#__PURE__*/React.createElement(Text, {
    style: {
      color: '#bbb'
    }
  }, "Block"));
};

var WingBlankExample = /*#__PURE__*/function (_React$Component) {
  _inherits(WingBlankExample, _React$Component);

  var _super = _createSuper(WingBlankExample);

  function WingBlankExample() {
    _classCallCheck(this, WingBlankExample);

    return _super.apply(this, arguments);
  }

  _createClass(WingBlankExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(WhiteSpace, null), /*#__PURE__*/React.createElement(WingBlank, null, /*#__PURE__*/React.createElement(PlaceHolder, null)), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(WingBlank, {
        size: "md"
      }, /*#__PURE__*/React.createElement(PlaceHolder, null)), /*#__PURE__*/React.createElement(WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/React.createElement(WingBlank, {
        size: "sm"
      }, /*#__PURE__*/React.createElement(PlaceHolder, null)));
    }
  }]);

  return WingBlankExample;
}(React.Component);

export { WingBlankExample as default };
