// @ant-design/react-native/lib/card/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

var BasicCardExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicCardExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicCardExample);

  function BasicCardExample() {
    (0, _classCallCheck2["default"])(this, BasicCardExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicCardExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          paddingTop: 30
        }
      }, /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.Card, null, /*#__PURE__*/_react["default"].createElement(_.Card.Header, {
        title: "This is title",
        thumbStyle: {
          width: 30,
          height: 30
        },
        thumb: "https://gw.alipayobjects.com/zos/rmsportal/MRhHctKOineMbKAZslML.jpg",
        extra: "this is extra"
      }), /*#__PURE__*/_react["default"].createElement(_.Card.Body, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          height: 42
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          marginLeft: 16
        }
      }, "Card Content"))), /*#__PURE__*/_react["default"].createElement(_.Card.Footer, {
        content: "footer content",
        extra: "footer extra content"
      }))), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Card, {
        full: true
      }, /*#__PURE__*/_react["default"].createElement(_.Card.Header, {
        title: "Full Column",
        thumbStyle: {
          width: 30,
          height: 30
        },
        thumb: "https://gw.alipayobjects.com/zos/rmsportal/MRhHctKOineMbKAZslML.jpg",
        extra: "this is extra"
      }), /*#__PURE__*/_react["default"].createElement(_.Card.Body, null, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          height: 42
        }
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
        style: {
          marginLeft: 16
        }
      }, "Card Content"))), /*#__PURE__*/_react["default"].createElement(_.Card.Footer, {
        content: "footer content",
        extra: "footer extra content"
      })));
    }
  }]);
  return BasicCardExample;
}(_react["default"].Component);

exports["default"] = BasicCardExample;
