// @ant-design/react-native/lib/pagination/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _ = require("../../");

var locale = {
  prevText: '上一步',
  nextText: '下一步'
};

var BasicPaginationExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicPaginationExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicPaginationExample);

  function BasicPaginationExample() {
    (0, _classCallCheck2["default"])(this, BasicPaginationExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicPaginationExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_.WingBlank, {
        size: "lg"
      }, /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Pagination, {
        total: 5,
        current: 1,
        locale: locale
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Pagination, {
        simple: true,
        total: 5,
        current: 1,
        locale: locale
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Pagination, {
        mode: "number",
        total: 5,
        current: 3
      }), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Pagination, {
        mode: "pointer",
        total: 5,
        current: 2
      }));
    }
  }]);
  return BasicPaginationExample;
}(_react["default"].Component);

exports["default"] = BasicPaginationExample;
