// @ant-design/react-native/es/tabs/demo/basic.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

/* tslint:disable:no-console */
import React from 'react';
import { ScrollView, Text, TouchableOpacity, View } from 'react-native';
import Tabs from '..';

var renderContent = function renderContent(tab, index) {
  var style = {
    paddingVertical: 40,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    backgroundColor: '#ddd'
  };
  var content = [1, 2, 3, 4, 5, 6, 7, 8].map(function (i) {
    return /*#__PURE__*/React.createElement(View, {
      key: "".concat(index, "_").concat(i),
      style: style
    }, /*#__PURE__*/React.createElement(Text, null, tab.title, " - ", i));
  });
  return /*#__PURE__*/React.createElement(ScrollView, {
    style: {
      backgroundColor: '#fff'
    }
  }, content);
};

var BasicTabsExample = /*#__PURE__*/function (_React$Component) {
  _inherits(BasicTabsExample, _React$Component);

  var _super = _createSuper(BasicTabsExample);

  function BasicTabsExample() {
    _classCallCheck(this, BasicTabsExample);

    return _super.apply(this, arguments);
  }

  _createClass(BasicTabsExample, [{
    key: "render",
    value: function render() {
      var tabs = [{
        title: 'First Tab'
      }, {
        title: 'Second Tab'
      }, {
        title: 'Third Tab'
      }];
      var tabs2 = [{
        title: '1st Tab'
      }, {
        title: '2nd Tab'
      }, {
        title: '3rd Tab'
      }, {
        title: '4th Tab'
      }, {
        title: '5th Tab'
      }, {
        title: '6th Tab'
      }, {
        title: '7th Tab'
      }, {
        title: '8th Tab'
      }, {
        title: '9th Tab'
      }];
      var style = {
        alignItems: 'center',
        justifyContent: 'center',
        height: 150,
        backgroundColor: '#fff'
      };
      return /*#__PURE__*/React.createElement(View, {
        style: {
          flex: 1
        }
      }, /*#__PURE__*/React.createElement(Tabs, {
        tabs: tabs
      }, /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of First Tab")), /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of Second Tab")), /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of Third Tab"))), /*#__PURE__*/React.createElement(Text, {
        style: {
          margin: 16
        }
      }, "Custom RenderTabBar"), /*#__PURE__*/React.createElement(Tabs, {
        tabs: tabs,
        renderTabBar: function renderTabBar(tabProps) {
          return /*#__PURE__*/React.createElement(View, {
            style: {
              paddingHorizontal: 16,
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'space-evenly'
            }
          }, tabProps.tabs.map(function (tab, i) {
            return (
              /*#__PURE__*/
              // change the style to fit your needs
              React.createElement(TouchableOpacity, {
                activeOpacity: 0.9,
                key: tab.key || i,
                style: {
                  // width: '30%',
                  padding: 6
                },
                onPress: function onPress() {
                  var goToTab = tabProps.goToTab,
                      onTabClick = tabProps.onTabClick; // tslint:disable-next-line:no-unused-expression

                  onTabClick && onTabClick(tabs[i], i); // tslint:disable-next-line:no-unused-expression

                  goToTab && goToTab(i);
                }
              }, /*#__PURE__*/React.createElement(Text, {
                style: {
                  color: tabProps.activeTab === i ? 'green' : undefined
                }
              }, tab.title))
            );
          }));
        }
      }, /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of First Tab")), /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of Second Tab")), /*#__PURE__*/React.createElement(View, {
        style: style
      }, /*#__PURE__*/React.createElement(Text, null, "Content of Third Tab"))), /*#__PURE__*/React.createElement(View, {
        style: {
          flex: 2
        }
      }, /*#__PURE__*/React.createElement(Tabs, {
        tabs: tabs2,
        initialPage: 1,
        tabBarPosition: "top"
      }, renderContent)));
    }
  }]);

  return BasicTabsExample;
}(React.Component);

export { BasicTabsExample as default };
export var title = 'Tabs';
export var description = 'Tabs example';
