// @ant-design/react-native/lib/radio/RadioItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _index = _interopRequireDefault(require("../list/index"));

var _style = require("../style");

var _Radio = _interopRequireDefault(require("./Radio"));

var _index2 = _interopRequireDefault(require("./style/index"));

var ListItem = _index["default"].Item;

var RadioItem = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(RadioItem, _React$Component);

  var _super = (0, _createSuper2["default"])(RadioItem);

  function RadioItem() {
    var _this;

    (0, _classCallCheck2["default"])(this, RadioItem);
    _this = _super.apply(this, arguments);

    _this.handleClick = function () {
      if (_this.radio) {
        _this.radio.handleClick();
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(RadioItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          radioStyle = _this$props.radioStyle,
          defaultChecked = _this$props.defaultChecked,
          checked = _this$props.checked,
          disabled = _this$props.disabled,
          children = _this$props.children,
          onChange = _this$props.onChange;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index2["default"]
      }, function (styles) {
        var contentDom = null;

        if (children && /*#__PURE__*/_react["default"].isValidElement(children)) {
          contentDom = /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
            style: {
              flex: 1
            }
          }, children);
        } else {
          var contentStyle = [styles.radioItemContent, disabled ? styles.radioItemContentDisable : {}];
          contentDom = /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
            style: contentStyle,
            numberOfLines: 1
          }, _this2.props.children);
        }

        var radioEl = /*#__PURE__*/_react["default"].createElement(_Radio["default"], {
          ref: function ref(_ref) {
            return _this2.radio = _ref;
          },
          style: radioStyle,
          defaultChecked: defaultChecked,
          checked: checked,
          onChange: onChange,
          disabled: disabled
        });

        return /*#__PURE__*/_react["default"].createElement(ListItem, {
          style: style,
          onPress: disabled ? undefined : _this2.handleClick,
          extra: radioEl
        }, contentDom);
      });
    }
  }]);
  return RadioItem;
}(_react["default"].Component);

exports["default"] = RadioItem;
