// @ant-design/react-native/lib/radio/Radio.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _icon = _interopRequireDefault(require("../icon"));

var _style = require("../style");

var _default = _interopRequireDefault(require("../style/themes/default"));

var _index = _interopRequireDefault(require("./style/index"));

var Radio = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Radio, _React$Component);

  var _super = (0, _createSuper2["default"])(Radio);

  function Radio(props, context) {
    var _this;

    (0, _classCallCheck2["default"])(this, Radio);
    _this = _super.call(this, props, context);

    _this.handleClick = function () {
      if (_this.props.disabled) {
        return;
      }

      if (!('checked' in _this.props)) {
        _this.setState({
          checked: true
        });
      }

      if (_this.props.onChange) {
        _this.props.onChange({
          target: {
            checked: true
          }
        });
      }
    };

    _this.state = {
      checked: props.checked || props.defaultChecked || false
    };
    return _this;
  }

  (0, _createClass2["default"])(Radio, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if ('checked' in nextProps) {
        this.setState({
          checked: !!nextProps.checked
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          style = _this$props.style,
          disabled = _this$props.disabled,
          children = _this$props.children;
      return /*#__PURE__*/_react["default"].createElement(_style.WithTheme, {
        styles: this.props.styles,
        themeStyles: _index["default"]
      }, function (styles) {
        var checked = _this2.state.checked;
        var icon;

        if (checked) {
          icon = disabled ? /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "check",
            style: [styles.icon, style]
          }) : /*#__PURE__*/_react["default"].createElement(_icon["default"], {
            name: "check",
            color: _default["default"].brand_primary,
            style: [styles.icon, style]
          });
        }

        return /*#__PURE__*/_react["default"].createElement(_reactNative.TouchableWithoutFeedback, {
          onPress: _this2.handleClick
        }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
          style: [styles.wrapper]
        }, icon, typeof children === 'string' ?
        /*#__PURE__*/
        // tslint:disable-next-line:jsx-no-multiline-js
        _react["default"].createElement(_reactNative.Text, null, _this2.props.children) : children));
      });
    }
  }]);
  return Radio;
}(_react["default"].Component);

exports["default"] = Radio;
