// @ant-design/react-native/lib/tag/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
function onChange(selected) {
  console.log("tag selected: ".concat(selected));
}

var BasicTagExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicTagExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicTagExample);

  function BasicTagExample() {
    (0, _classCallCheck2["default"])(this, BasicTagExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicTagExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          padding: 10
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Tag, null, "Basic"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        selected: true
      }, "Selected"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        disabled: true
      }, "Disabled"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        onChange: onChange
      }, "Callback"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        closable: true,
        onClose: function onClose() {
          console.log('onClose');
        },
        afterClose: function afterClose() {
          console.log('afterClose');
        }
      }, "Closable"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        small: true
      }, "Small and Readonly"), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, null), /*#__PURE__*/_react["default"].createElement(_.Tag, {
        onLongPress: function onLongPress() {
          console.log('onLongPress');
        }
      }, "LongPress"));
    }
  }]);
  return BasicTagExample;
}(_react["default"].Component);

exports["default"] = BasicTagExample;
