// @ant-design/react-native/lib/notice-bar/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var NoticeBarExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(NoticeBarExample, _React$Component);

  var _super = (0, _createSuper2["default"])(NoticeBarExample);

  function NoticeBarExample() {
    (0, _classCallCheck2["default"])(this, NoticeBarExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(NoticeBarExample, [{
    key: "render",
    value: function render() {
      var customIcon = /*#__PURE__*/_react["default"].createElement(_reactNative.Image // tslint:disable-next-line:jsx-no-multiline-js
      , {
        // tslint:disable-next-line:jsx-no-multiline-js
        source: {
          uri: 'https://zos.alipayobjects.com/rmsportal/bRnouywfdRsCcLU.png'
        },
        style: {
          width: 12,
          height: 12
        }
      });

      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 10
        }
      }, /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        onPress: function onPress() {
          return alert('click');
        },
        marqueeProps: {
          loop: true,
          style: {
            fontSize: 12,
            color: 'red'
          }
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "closable",
        onPress: function onPress() {
          return alert('will close');
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "closable",
        icon: customIcon
      }, "Customized icon."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "link",
        onPress: function onPress() {
          return alert('link');
        }
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "link"
      }, "Notice: The arrival time of incomes and transfers of Yu 'E Bao will be delayed during National Day."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "closable",
        icon: undefined
      }, "Remove the default icon."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "closable",
        action: /*#__PURE__*/_react["default"].createElement(_reactNative.Text, {
          style: {
            color: '#a1a1a1'
          }
        }, "\u4E0D\u518D\u63D0\u793A")
      }, "Closable demo for `actionText`."), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.NoticeBar, {
        mode: "link",
        action: /*#__PURE__*/_react["default"].createElement(_reactNative.Text, null, "\u53BB\u770B\u770B")
      }, "Link demo for `actionText`."));
    }
  }]);
  return NoticeBarExample;
}(_react["default"].Component);

exports["default"] = NoticeBarExample;
