// @ant-design/react-native/lib/accordion/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var AccordionExmple = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(AccordionExmple, _React$Component);

  var _super = (0, _createSuper2["default"])(AccordionExmple);

  function AccordionExmple() {
    var _this;

    (0, _classCallCheck2["default"])(this, AccordionExmple);
    _this = _super.apply(this, arguments);
    _this.state = {
      activeSections: [2, 0]
    };

    _this.onChange = function (activeSections) {
      _this.setState({
        activeSections: activeSections
      });
    };

    return _this;
  }

  (0, _createClass2["default"])(AccordionExmple, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          marginTop: 80,
          marginBottom: 10
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Accordion, {
        onChange: this.onChange,
        activeSections: this.state.activeSections
      }, /*#__PURE__*/_react["default"].createElement(_.Accordion.Panel, {
        header: "Title 1"
      }, /*#__PURE__*/_react["default"].createElement(_.List, null, /*#__PURE__*/_react["default"].createElement(_.List.Item, null, "Content 1"), /*#__PURE__*/_react["default"].createElement(_.List.Item, null, "Content 2"), /*#__PURE__*/_react["default"].createElement(_.List.Item, null, "Content 3"))), /*#__PURE__*/_react["default"].createElement(_.Accordion.Panel, {
        header: "Title 2"
      }, "this is panel content2 or other"), /*#__PURE__*/_react["default"].createElement(_.Accordion.Panel, {
        header: "Title 3"
      }, "Text text text text text text text text text text text text text text text")));
    }
  }]);
  return AccordionExmple;
}(_react["default"].Component);

exports["default"] = AccordionExmple;
