// @ant-design/react-native/lib/badge/demo/basic.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _ = require("../../");

/* tslint:disable:no-console */
var BasicTagExample = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(BasicTagExample, _React$Component);

  var _super = (0, _createSuper2["default"])(BasicTagExample);

  function BasicTagExample() {
    (0, _classCallCheck2["default"])(this, BasicTagExample);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(BasicTagExample, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactNative.ScrollView, {
        style: {
          flex: 1
        },
        automaticallyAdjustContentInsets: false,
        showsHorizontalScrollIndicator: false,
        showsVerticalScrollIndicator: false
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          padding: 20
        }
      }, /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: 9
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: 109,
        overflowCount: 100
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: 109
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: "new"
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: 109,
        dot: true
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      })), /*#__PURE__*/_react["default"].createElement(_.WhiteSpace, {
        size: "lg"
      }), /*#__PURE__*/_react["default"].createElement(_.Badge, {
        text: 33,
        corner: true
      }, /*#__PURE__*/_react["default"].createElement(_reactNative.View, {
        style: {
          width: 52,
          height: 52,
          backgroundColor: 'rgba(255, 140, 101, 0.15)'
        }
      }))));
    }
  }]);
  return BasicTagExample;
}(_react["default"].Component);

exports["default"] = BasicTagExample;
